<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">@import url('https://themes.googleusercontent.com/fonts/css?kit=cGvuclDC_Z1vE_cnVEU6AaCTwcgESYH44X24siBZF_L3hqPTPW_VOAb4EcAVF3PNVbTOx7nbtm8DB2MS2CCfo1D3_G7kCgzfrJfr6t4JH-JHLRYT7TsASg17ok0VYewe16z_4oSSXTKe9SG2liOzJA');ol.lst-kix_odlzlhcdt35z-2.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-2 0}.lst-kix_xzg29ra0f9c3-0>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-0,decimal) ". "}.lst-kix_xzg29ra0f9c3-1>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-1,lower-latin) ". "}.lst-kix_xzg29ra0f9c3-2>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-2}ol.lst-kix_o260vi9n67px-1{list-style-type:none}ol.lst-kix_o260vi9n67px-2{list-style-type:none}ol.lst-kix_o260vi9n67px-3{list-style-type:none}ol.lst-kix_o260vi9n67px-4{list-style-type:none}.lst-kix_xzg29ra0f9c3-6>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-6,decimal) ". "}ol.lst-kix_o260vi9n67px-5{list-style-type:none}ol.lst-kix_o260vi9n67px-6{list-style-type:none}ol.lst-kix_o260vi9n67px-7{list-style-type:none}ol.lst-kix_o260vi9n67px-8{list-style-type:none}.lst-kix_xzg29ra0f9c3-5>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-5,lower-roman) ". "}.lst-kix_o260vi9n67px-1>li{counter-increment:lst-ctn-kix_o260vi9n67px-1}.lst-kix_xzg29ra0f9c3-2>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-2,lower-roman) ". "}.lst-kix_xzg29ra0f9c3-4>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-4,lower-latin) ". "}ol.lst-kix_xzg29ra0f9c3-3.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-3 0}ol.lst-kix_o260vi9n67px-0{list-style-type:none}.lst-kix_xzg29ra0f9c3-3>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-3,decimal) ". "}ol.lst-kix_odlzlhcdt35z-8.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-8 0}ol.lst-kix_o260vi9n67px-4.start{counter-reset:lst-ctn-kix_o260vi9n67px-4 0}.lst-kix_cua9hof9g9u-7>li{counter-increment:lst-ctn-kix_cua9hof9g9u-7}ol.lst-kix_7b6g4smvhj31-5.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-5 0}.lst-kix_o260vi9n67px-3>li{counter-increment:lst-ctn-kix_o260vi9n67px-3}.lst-kix_odlzlhcdt35z-1>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-1}ol.lst-kix_cua9hof9g9u-5.start{counter-reset:lst-ctn-kix_cua9hof9g9u-5 0}.lst-kix_o260vi9n67px-5>li{counter-increment:lst-ctn-kix_o260vi9n67px-5}.lst-kix_cua9hof9g9u-0>li{counter-increment:lst-ctn-kix_cua9hof9g9u-0}.lst-kix_ur78f3m641k9-1>li:before{content:"\0025cb  "}.lst-kix_xzg29ra0f9c3-0>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-0}.lst-kix_ur78f3m641k9-2>li:before{content:"\0025a0  "}ol.lst-kix_odlzlhcdt35z-7.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-7 0}.lst-kix_ur78f3m641k9-3>li:before{content:"\0025cf  "}.lst-kix_7b6g4smvhj31-2>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-2}ol.lst-kix_o260vi9n67px-3.start{counter-reset:lst-ctn-kix_o260vi9n67px-3 0}.lst-kix_xzg29ra0f9c3-6>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-6}ol.lst-kix_cua9hof9g9u-4.start{counter-reset:lst-ctn-kix_cua9hof9g9u-4 0}ol.lst-kix_7b6g4smvhj31-6.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-6 0}.lst-kix_ur78f3m641k9-0>li:before{content:"\0025cf  "}.lst-kix_odlzlhcdt35z-3>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-3}ul.lst-kix_6c526n8mx7cb-0{list-style-type:none}ul.lst-kix_6c526n8mx7cb-1{list-style-type:none}ul.lst-kix_6c526n8mx7cb-2{list-style-type:none}ul.lst-kix_6c526n8mx7cb-3{list-style-type:none}ul.lst-kix_6c526n8mx7cb-4{list-style-type:none}ul.lst-kix_r5gz416xmm5m-0{list-style-type:none}ul.lst-kix_6c526n8mx7cb-5{list-style-type:none}ul.lst-kix_r5gz416xmm5m-1{list-style-type:none}ul.lst-kix_6c526n8mx7cb-6{list-style-type:none}ul.lst-kix_r5gz416xmm5m-2{list-style-type:none}ul.lst-kix_6c526n8mx7cb-7{list-style-type:none}ul.lst-kix_r5gz416xmm5m-3{list-style-type:none}ul.lst-kix_6c526n8mx7cb-8{list-style-type:none}ol.lst-kix_o260vi9n67px-8.start{counter-reset:lst-ctn-kix_o260vi9n67px-8 0}.lst-kix_xzg29ra0f9c3-7>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-7,lower-latin) ". "}ul.lst-kix_r5gz416xmm5m-8{list-style-type:none}ul.lst-kix_nmlk7fvtpy78-6{list-style-type:none}ul.lst-kix_nmlk7fvtpy78-7{list-style-type:none}ul.lst-kix_nmlk7fvtpy78-4{list-style-type:none}.lst-kix_ur78f3m641k9-4>li:before{content:"\0025cb  "}.lst-kix_ur78f3m641k9-6>li:before{content:"\0025cf  "}ul.lst-kix_nmlk7fvtpy78-5{list-style-type:none}.lst-kix_xzg29ra0f9c3-8>li:before{content:"" counter(lst-ctn-kix_xzg29ra0f9c3-8,lower-roman) ". "}ul.lst-kix_r5gz416xmm5m-4{list-style-type:none}ul.lst-kix_r5gz416xmm5m-5{list-style-type:none}ul.lst-kix_r5gz416xmm5m-6{list-style-type:none}ul.lst-kix_nmlk7fvtpy78-8{list-style-type:none}.lst-kix_ur78f3m641k9-5>li:before{content:"\0025a0  "}ul.lst-kix_r5gz416xmm5m-7{list-style-type:none}ol.lst-kix_7b6g4smvhj31-0.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-0 0}.lst-kix_ur78f3m641k9-8>li:before{content:"\0025a0  "}ul.lst-kix_nmlk7fvtpy78-2{list-style-type:none}ul.lst-kix_nmlk7fvtpy78-3{list-style-type:none}ul.lst-kix_nmlk7fvtpy78-0{list-style-type:none}.lst-kix_ur78f3m641k9-7>li:before{content:"\0025cb  "}ul.lst-kix_nmlk7fvtpy78-1{list-style-type:none}.lst-kix_7b6g4smvhj31-2>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-2,lower-roman) ". "}.lst-kix_7b6g4smvhj31-4>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-4,lower-latin) ". "}ol.lst-kix_xzg29ra0f9c3-7{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-8{list-style-type:none}.lst-kix_o260vi9n67px-8>li{counter-increment:lst-ctn-kix_o260vi9n67px-8}.lst-kix_odlzlhcdt35z-7>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-7}ol.lst-kix_xzg29ra0f9c3-1{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-2{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-0{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-5{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-6{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-3{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-4{list-style-type:none}.lst-kix_rxkpu1ydcip2-1>li:before{content:"\0025cb  "}.lst-kix_rxkpu1ydcip2-3>li:before{content:"\0025cf  "}.lst-kix_o260vi9n67px-7>li{counter-increment:lst-ctn-kix_o260vi9n67px-7}.lst-kix_7b6g4smvhj31-0>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-0,decimal) ". "}ol.lst-kix_cua9hof9g9u-1.start{counter-reset:lst-ctn-kix_cua9hof9g9u-1 0}ol.lst-kix_xzg29ra0f9c3-7.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-7 0}.lst-kix_cua9hof9g9u-4>li{counter-increment:lst-ctn-kix_cua9hof9g9u-4}ol.lst-kix_odlzlhcdt35z-4.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-4 0}.lst-kix_7b6g4smvhj31-4>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-4}.lst-kix_o260vi9n67px-6>li{counter-increment:lst-ctn-kix_o260vi9n67px-6}ol.lst-kix_odlzlhcdt35z-1.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-1 0}.lst-kix_cua9hof9g9u-5>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-5,lower-roman) ". "}.lst-kix_r5gz416xmm5m-1>li:before{content:"\0025cb  "}.lst-kix_cua9hof9g9u-3>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-3,decimal) ". "}ol.lst-kix_xzg29ra0f9c3-8.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-8 0}.lst-kix_cua9hof9g9u-1>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-1,lower-latin) ". "}.lst-kix_nmlk7fvtpy78-3>li:before{content:"\0025cf  "}.lst-kix_xzg29ra0f9c3-4>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-4}.lst-kix_7b6g4smvhj31-3>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-3}.lst-kix_nmlk7fvtpy78-1>li:before{content:"\0025cb  "}.lst-kix_nmlk7fvtpy78-5>li:before{content:"\0025a0  "}.lst-kix_87rvu8ltvc9n-4>li:before{content:"\0025cb  "}.lst-kix_87rvu8ltvc9n-8>li:before{content:"\0025a0  "}.lst-kix_rxkpu1ydcip2-5>li:before{content:"\0025a0  "}ol.lst-kix_odlzlhcdt35z-3.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-3 0}.lst-kix_87rvu8ltvc9n-6>li:before{content:"\0025cf  "}.lst-kix_6c526n8mx7cb-3>li:before{content:"\0025cf  "}.lst-kix_r5gz416xmm5m-3>li:before{content:"\0025cf  "}.lst-kix_6c526n8mx7cb-5>li:before{content:"\0025a0  "}.lst-kix_odlzlhcdt35z-8>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-8}.lst-kix_cua9hof9g9u-5>li{counter-increment:lst-ctn-kix_cua9hof9g9u-5}.lst-kix_6c526n8mx7cb-1>li:before{content:"\0025cb  "}.lst-kix_nmlk7fvtpy78-7>li:before{content:"\0025cb  "}ol.lst-kix_o260vi9n67px-5.start{counter-reset:lst-ctn-kix_o260vi9n67px-5 0}.lst-kix_rxkpu1ydcip2-7>li:before{content:"\0025cb  "}.lst-kix_7b6g4smvhj31-6>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-6,decimal) ". "}.lst-kix_7b6g4smvhj31-8>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-8,lower-roman) ". "}ol.lst-kix_cua9hof9g9u-0.start{counter-reset:lst-ctn-kix_cua9hof9g9u-0 0}.lst-kix_r5gz416xmm5m-5>li:before{content:"\0025a0  "}.lst-kix_r5gz416xmm5m-7>li:before{content:"\0025cb  "}ol.lst-kix_7b6g4smvhj31-8.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-8 0}.lst-kix_o260vi9n67px-2>li{counter-increment:lst-ctn-kix_o260vi9n67px-2}ol.lst-kix_xzg29ra0f9c3-6.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-6 0}ol.lst-kix_cua9hof9g9u-2.start{counter-reset:lst-ctn-kix_cua9hof9g9u-2 0}.lst-kix_6c526n8mx7cb-7>li:before{content:"\0025cb  "}.lst-kix_6c526n8mx7cb-6>li:before{content:"\0025cf  "}.lst-kix_6c526n8mx7cb-8>li:before{content:"\0025a0  "}.lst-kix_odlzlhcdt35z-2>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-2}ol.lst-kix_odlzlhcdt35z-5.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-5 0}.lst-kix_87rvu8ltvc9n-0>li:before{content:"\0025cf  "}.lst-kix_xzg29ra0f9c3-3>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-3}ol.lst-kix_7b6g4smvhj31-3{list-style-type:none}.lst-kix_87rvu8ltvc9n-1>li:before{content:"\0025cb  "}ol.lst-kix_7b6g4smvhj31-4{list-style-type:none}ol.lst-kix_7b6g4smvhj31-5{list-style-type:none}ol.lst-kix_7b6g4smvhj31-6{list-style-type:none}ol.lst-kix_7b6g4smvhj31-7{list-style-type:none}ol.lst-kix_o260vi9n67px-1.start{counter-reset:lst-ctn-kix_o260vi9n67px-1 0}.lst-kix_87rvu8ltvc9n-3>li:before{content:"\0025cf  "}ol.lst-kix_7b6g4smvhj31-8{list-style-type:none}.lst-kix_87rvu8ltvc9n-2>li:before{content:"\0025a0  "}ol.lst-kix_o260vi9n67px-6.start{counter-reset:lst-ctn-kix_o260vi9n67px-6 0}ul.lst-kix_ur78f3m641k9-1{list-style-type:none}ul.lst-kix_ur78f3m641k9-0{list-style-type:none}ol.lst-kix_cua9hof9g9u-8.start{counter-reset:lst-ctn-kix_cua9hof9g9u-8 0}ol.lst-kix_7b6g4smvhj31-0{list-style-type:none}ol.lst-kix_7b6g4smvhj31-1{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-0.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-0 0}ol.lst-kix_7b6g4smvhj31-2{list-style-type:none}ol.lst-kix_7b6g4smvhj31-2.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-2 0}.lst-kix_cua9hof9g9u-6>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-6,decimal) ". "}.lst-kix_o260vi9n67px-0>li{counter-increment:lst-ctn-kix_o260vi9n67px-0}.lst-kix_cua9hof9g9u-7>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-7,lower-latin) ". "}.lst-kix_odlzlhcdt35z-4>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-4}.lst-kix_7b6g4smvhj31-1>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-1}.lst-kix_cua9hof9g9u-8>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-8,lower-roman) ". "}.lst-kix_6meiqxvvvyw3-6>li:before{content:"\0025cf  "}.lst-kix_6meiqxvvvyw3-7>li:before{content:"\0025cb  "}.lst-kix_odlzlhcdt35z-5>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-5,lower-roman) ". "}.lst-kix_odlzlhcdt35z-6>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-6,decimal) ". "}.lst-kix_o260vi9n67px-3>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-3,decimal) ". "}.lst-kix_o260vi9n67px-4>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-4,lower-latin) ". "}.lst-kix_6aptd0j42w1q-5>li:before{content:"\0025a0  "}.lst-kix_6aptd0j42w1q-7>li:before{content:"\0025cb  "}.lst-kix_odlzlhcdt35z-7>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-7,lower-latin) ". "}ol.lst-kix_o260vi9n67px-7.start{counter-reset:lst-ctn-kix_o260vi9n67px-7 0}.lst-kix_6meiqxvvvyw3-4>li:before{content:"\0025cb  "}.lst-kix_6meiqxvvvyw3-5>li:before{content:"\0025a0  "}.lst-kix_6meiqxvvvyw3-8>li:before{content:"\0025a0  "}.lst-kix_odlzlhcdt35z-3>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-3,decimal) ". "}.lst-kix_o260vi9n67px-1>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-1,lower-latin) ". "}.lst-kix_o260vi9n67px-5>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-5,lower-roman) ". "}.lst-kix_6aptd0j42w1q-4>li:before{content:"\0025cb  "}.lst-kix_6aptd0j42w1q-8>li:before{content:"\0025a0  "}ul.lst-kix_6aptd0j42w1q-3{list-style-type:none}ul.lst-kix_6aptd0j42w1q-4{list-style-type:none}.lst-kix_o260vi9n67px-0>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-0,decimal) ". "}.lst-kix_o260vi9n67px-7>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-7,lower-latin) ". "}.lst-kix_o260vi9n67px-8>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-8,lower-roman) ". "}ul.lst-kix_6aptd0j42w1q-5{list-style-type:none}.lst-kix_6aptd0j42w1q-1>li:before{content:"\0025cb  "}.lst-kix_6aptd0j42w1q-3>li:before{content:"\0025cf  "}ul.lst-kix_6aptd0j42w1q-6{list-style-type:none}ul.lst-kix_6aptd0j42w1q-0{list-style-type:none}.lst-kix_odlzlhcdt35z-4>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-4,lower-latin) ". "}.lst-kix_o260vi9n67px-6>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-6,decimal) ". "}ul.lst-kix_6aptd0j42w1q-1{list-style-type:none}.lst-kix_6aptd0j42w1q-2>li:before{content:"\0025a0  "}ul.lst-kix_6aptd0j42w1q-2{list-style-type:none}.lst-kix_xzg29ra0f9c3-5>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-5}.lst-kix_6meiqxvvvyw3-0>li:before{content:"\0025cf  "}ul.lst-kix_6meiqxvvvyw3-8{list-style-type:none}ul.lst-kix_6aptd0j42w1q-7{list-style-type:none}.lst-kix_6meiqxvvvyw3-1>li:before{content:"\0025cb  "}ul.lst-kix_6aptd0j42w1q-8{list-style-type:none}ol.lst-kix_odlzlhcdt35z-0.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-0 0}.lst-kix_6aptd0j42w1q-0>li:before{content:"\0025cf  "}ul.lst-kix_6meiqxvvvyw3-2{list-style-type:none}.lst-kix_6meiqxvvvyw3-3>li:before{content:"\0025cf  "}.lst-kix_cua9hof9g9u-6>li{counter-increment:lst-ctn-kix_cua9hof9g9u-6}ul.lst-kix_6meiqxvvvyw3-3{list-style-type:none}.lst-kix_odlzlhcdt35z-1>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-1,lower-latin) ". "}.lst-kix_odlzlhcdt35z-2>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-2,lower-roman) ". "}ul.lst-kix_6meiqxvvvyw3-0{list-style-type:none}.lst-kix_6meiqxvvvyw3-2>li:before{content:"\0025a0  "}ul.lst-kix_6meiqxvvvyw3-1{list-style-type:none}ul.lst-kix_6meiqxvvvyw3-6{list-style-type:none}ul.lst-kix_6meiqxvvvyw3-7{list-style-type:none}.lst-kix_odlzlhcdt35z-0>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-0,decimal) ". "}ul.lst-kix_6meiqxvvvyw3-4{list-style-type:none}ul.lst-kix_6meiqxvvvyw3-5{list-style-type:none}.lst-kix_odlzlhcdt35z-0>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-0}ol.lst-kix_xzg29ra0f9c3-5.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-5 0}ol.lst-kix_cua9hof9g9u-0{list-style-type:none}ol.lst-kix_cua9hof9g9u-1{list-style-type:none}ol.lst-kix_cua9hof9g9u-2{list-style-type:none}ol.lst-kix_7b6g4smvhj31-7.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-7 0}ol.lst-kix_cua9hof9g9u-3{list-style-type:none}ol.lst-kix_cua9hof9g9u-3.start{counter-reset:lst-ctn-kix_cua9hof9g9u-3 0}ol.lst-kix_cua9hof9g9u-8{list-style-type:none}.lst-kix_7b6g4smvhj31-8>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-8}ol.lst-kix_cua9hof9g9u-4{list-style-type:none}ol.lst-kix_cua9hof9g9u-5{list-style-type:none}ol.lst-kix_cua9hof9g9u-6{list-style-type:none}ol.lst-kix_cua9hof9g9u-7{list-style-type:none}ol.lst-kix_o260vi9n67px-2.start{counter-reset:lst-ctn-kix_o260vi9n67px-2 0}ol.lst-kix_odlzlhcdt35z-6.start{counter-reset:lst-ctn-kix_odlzlhcdt35z-6 0}.lst-kix_o260vi9n67px-2>li:before{content:"" counter(lst-ctn-kix_o260vi9n67px-2,lower-roman) ". "}.lst-kix_odlzlhcdt35z-8>li:before{content:"" counter(lst-ctn-kix_odlzlhcdt35z-8,lower-roman) ". "}.lst-kix_6aptd0j42w1q-6>li:before{content:"\0025cf  "}.lst-kix_7b6g4smvhj31-3>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-3,decimal) ". "}.lst-kix_7b6g4smvhj31-1>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-1,lower-latin) ". "}.lst-kix_7b6g4smvhj31-5>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-5,lower-roman) ". "}ol.lst-kix_odlzlhcdt35z-8{list-style-type:none}.lst-kix_xzg29ra0f9c3-8>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-8}ol.lst-kix_odlzlhcdt35z-7{list-style-type:none}ol.lst-kix_odlzlhcdt35z-4{list-style-type:none}ol.lst-kix_odlzlhcdt35z-3{list-style-type:none}ol.lst-kix_odlzlhcdt35z-6{list-style-type:none}.lst-kix_7b6g4smvhj31-5>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-5}ol.lst-kix_odlzlhcdt35z-5{list-style-type:none}.lst-kix_rxkpu1ydcip2-2>li:before{content:"\0025a0  "}ol.lst-kix_7b6g4smvhj31-4.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-4 0}.lst-kix_rxkpu1ydcip2-0>li:before{content:"\0025cf  "}ol.lst-kix_7b6g4smvhj31-1.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-1 0}.lst-kix_odlzlhcdt35z-6>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-6}.lst-kix_cua9hof9g9u-3>li{counter-increment:lst-ctn-kix_cua9hof9g9u-3}ul.lst-kix_87rvu8ltvc9n-7{list-style-type:none}ul.lst-kix_87rvu8ltvc9n-8{list-style-type:none}ul.lst-kix_87rvu8ltvc9n-5{list-style-type:none}ul.lst-kix_87rvu8ltvc9n-6{list-style-type:none}ul.lst-kix_87rvu8ltvc9n-3{list-style-type:none}ul.lst-kix_87rvu8ltvc9n-4{list-style-type:none}.lst-kix_cua9hof9g9u-1>li{counter-increment:lst-ctn-kix_cua9hof9g9u-1}ul.lst-kix_87rvu8ltvc9n-1{list-style-type:none}ul.lst-kix_87rvu8ltvc9n-2{list-style-type:none}ol.lst-kix_xzg29ra0f9c3-4.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-4 0}ul.lst-kix_87rvu8ltvc9n-0{list-style-type:none}ol.lst-kix_o260vi9n67px-0.start{counter-reset:lst-ctn-kix_o260vi9n67px-0 0}ol.lst-kix_cua9hof9g9u-7.start{counter-reset:lst-ctn-kix_cua9hof9g9u-7 0}.lst-kix_7b6g4smvhj31-7>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-7}.lst-kix_r5gz416xmm5m-8>li:before{content:"\0025a0  "}.lst-kix_odlzlhcdt35z-5>li{counter-increment:lst-ctn-kix_odlzlhcdt35z-5}ul.lst-kix_rxkpu1ydcip2-8{list-style-type:none}ol.lst-kix_7b6g4smvhj31-3.start{counter-reset:lst-ctn-kix_7b6g4smvhj31-3 0}ul.lst-kix_rxkpu1ydcip2-7{list-style-type:none}.lst-kix_r5gz416xmm5m-2>li:before{content:"\0025a0  "}ol.lst-kix_xzg29ra0f9c3-1.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-1 0}.lst-kix_cua9hof9g9u-4>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-4,lower-latin) ". "}.lst-kix_r5gz416xmm5m-0>li:before{content:"\0025cf  "}.lst-kix_o260vi9n67px-4>li{counter-increment:lst-ctn-kix_o260vi9n67px-4}ul.lst-kix_ur78f3m641k9-7{list-style-type:none}ul.lst-kix_ur78f3m641k9-6{list-style-type:none}.lst-kix_cua9hof9g9u-0>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-0,decimal) ". "}ul.lst-kix_ur78f3m641k9-8{list-style-type:none}.lst-kix_7b6g4smvhj31-6>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-6}ul.lst-kix_ur78f3m641k9-3{list-style-type:none}ul.lst-kix_ur78f3m641k9-2{list-style-type:none}.lst-kix_cua9hof9g9u-2>li:before{content:"" counter(lst-ctn-kix_cua9hof9g9u-2,lower-roman) ". "}ul.lst-kix_ur78f3m641k9-5{list-style-type:none}ul.lst-kix_ur78f3m641k9-4{list-style-type:none}.lst-kix_87rvu8ltvc9n-5>li:before{content:"\0025a0  "}.lst-kix_nmlk7fvtpy78-2>li:before{content:"\0025a0  "}.lst-kix_7b6g4smvhj31-0>li{counter-increment:lst-ctn-kix_7b6g4smvhj31-0}.lst-kix_nmlk7fvtpy78-0>li:before{content:"\0025cf  "}.lst-kix_nmlk7fvtpy78-4>li:before{content:"\0025cb  "}.lst-kix_xzg29ra0f9c3-7>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-7}ol.lst-kix_cua9hof9g9u-6.start{counter-reset:lst-ctn-kix_cua9hof9g9u-6 0}ol.lst-kix_xzg29ra0f9c3-2.start{counter-reset:lst-ctn-kix_xzg29ra0f9c3-2 0}.lst-kix_6c526n8mx7cb-0>li:before{content:"\0025cf  "}.lst-kix_rxkpu1ydcip2-4>li:before{content:"\0025cb  "}.lst-kix_87rvu8ltvc9n-7>li:before{content:"\0025cb  "}li.li-bullet-0:before{margin-left:-18pt;white-space:nowrap;display:inline-block;min-width:18pt}ul.lst-kix_rxkpu1ydcip2-6{list-style-type:none}ol.lst-kix_odlzlhcdt35z-0{list-style-type:none}.lst-kix_cua9hof9g9u-2>li{counter-increment:lst-ctn-kix_cua9hof9g9u-2}ul.lst-kix_rxkpu1ydcip2-5{list-style-type:none}ul.lst-kix_rxkpu1ydcip2-4{list-style-type:none}.lst-kix_6c526n8mx7cb-2>li:before{content:"\0025a0  "}.lst-kix_6c526n8mx7cb-4>li:before{content:"\0025cb  "}ol.lst-kix_odlzlhcdt35z-2{list-style-type:none}ul.lst-kix_rxkpu1ydcip2-3{list-style-type:none}ol.lst-kix_odlzlhcdt35z-1{list-style-type:none}ul.lst-kix_rxkpu1ydcip2-2{list-style-type:none}.lst-kix_cua9hof9g9u-8>li{counter-increment:lst-ctn-kix_cua9hof9g9u-8}ul.lst-kix_rxkpu1ydcip2-1{list-style-type:none}.lst-kix_nmlk7fvtpy78-8>li:before{content:"\0025a0  "}ul.lst-kix_rxkpu1ydcip2-0{list-style-type:none}.lst-kix_r5gz416xmm5m-4>li:before{content:"\0025cb  "}.lst-kix_rxkpu1ydcip2-6>li:before{content:"\0025cf  "}.lst-kix_7b6g4smvhj31-7>li:before{content:"" counter(lst-ctn-kix_7b6g4smvhj31-7,lower-latin) ". "}.lst-kix_nmlk7fvtpy78-6>li:before{content:"\0025cf  "}.lst-kix_rxkpu1ydcip2-8>li:before{content:"\0025a0  "}.lst-kix_r5gz416xmm5m-6>li:before{content:"\0025cf  "}.lst-kix_xzg29ra0f9c3-1>li{counter-increment:lst-ctn-kix_xzg29ra0f9c3-1}ol{margin:0;padding:0}table td,table th{padding:0}.c62{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#b7b7b7;border-top-width:1pt;border-right-width:1pt;border-left-color:#b7b7b7;vertical-align:top;border-right-color:#b7b7b7;border-left-width:1pt;border-top-style:solid;background-color:#d9d9d9;border-left-style:solid;border-bottom-width:1pt;width:64.5pt;border-top-color:#b7b7b7;border-bottom-style:solid}.c35{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#b7b7b7;border-top-width:1pt;border-right-width:1pt;border-left-color:#b7b7b7;vertical-align:top;border-right-color:#b7b7b7;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:59.2pt;border-top-color:#b7b7b7;border-bottom-style:solid}.c47{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#b7b7b7;border-top-width:1pt;border-right-width:1pt;border-left-color:#b7b7b7;vertical-align:top;border-right-color:#b7b7b7;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:51pt;border-top-color:#b7b7b7;border-bottom-style:solid}.c77{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#b7b7b7;border-top-width:1pt;border-right-width:1pt;border-left-color:#b7b7b7;vertical-align:top;border-right-color:#b7b7b7;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:64.5pt;border-top-color:#b7b7b7;border-bottom-style:solid}.c50{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#b7b7b7;border-top-width:1pt;border-right-width:1pt;border-left-color:#b7b7b7;vertical-align:top;border-right-color:#b7b7b7;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:72pt;border-top-color:#b7b7b7;border-bottom-style:solid}.c36{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#b7b7b7;border-top-width:1pt;border-right-width:1pt;border-left-color:#b7b7b7;vertical-align:top;border-right-color:#b7b7b7;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:58.5pt;border-top-color:#b7b7b7;border-bottom-style:solid}.c42{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#ffffff;border-top-width:1pt;border-right-width:1pt;border-left-color:#ffffff;vertical-align:top;border-right-color:#ffffff;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:318pt;border-top-color:#ffffff;border-bottom-style:solid}.c69{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#b7b7b7;border-top-width:1pt;border-right-width:1pt;border-left-color:#b7b7b7;vertical-align:top;border-right-color:#b7b7b7;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:71.2pt;border-top-color:#b7b7b7;border-bottom-style:solid}.c1{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#ffffff;border-top-width:1pt;border-right-width:1pt;border-left-color:#ffffff;vertical-align:top;border-right-color:#ffffff;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:150pt;border-top-color:#ffffff;border-bottom-style:solid}.c6{color:#666666;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:24pt;font-family:"Philosopher";font-style:normal}.c44{font-weight:400;vertical-align:baseline;font-size:11pt;font-family:"Droid Serif";font-style:normal}.c29{font-weight:400;vertical-align:baseline;font-size:9pt;font-family:"Pontano Sans";font-style:normal}.c41{font-weight:400;vertical-align:baseline;font-size:10pt;font-family:"Pontano Sans";font-style:normal}.c12{padding-top:10pt;padding-bottom:0pt;line-height:1.5;text-align:justify;height:24pt}.c18{padding-top:0pt;padding-bottom:0pt;line-height:1.5;text-align:center;height:12pt}.c10{font-weight:400;vertical-align:baseline;font-size:9pt;font-family:"Inconsolata";font-style:normal}.c11{font-weight:400;vertical-align:baseline;font-size:11pt;font-family:"Consolas";font-style:normal}.c19{font-weight:400;vertical-align:baseline;font-size:12pt;font-family:"Consolas";font-style:normal}.c5{background-color:#ffe599;font-size:12pt;font-family:"Droid Sans";color:#000000;font-weight:400}.c26{font-weight:400;vertical-align:baseline;font-size:10pt;font-family:"Consolas";font-style:normal}.c22{font-weight:400;vertical-align:baseline;font-size:18pt;font-family:"Philosopher";font-style:normal}.c14{font-weight:400;vertical-align:baseline;font-size:9pt;font-family:"Consolas";font-style:normal}.c8{font-weight:400;vertical-align:baseline;font-size:12pt;font-family:"Pontano Sans";font-style:normal}.c20{padding-top:10pt;padding-bottom:0pt;line-height:1.5;text-align:justify}.c13{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#ff0000;text-decoration:underline}.c71{padding-top:0pt;padding-bottom:8pt;line-height:1.2272727272727273;text-align:left}.c60{padding-top:0pt;padding-bottom:0pt;line-height:1.15;text-align:center}.c53{border-spacing:0;border-collapse:collapse;margin-right:auto}.c65{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right}.c24{padding-top:0pt;padding-bottom:0pt;line-height:1.5;text-align:center}.c34{padding-top:0pt;padding-bottom:0pt;line-height:1.15;text-align:justify}.c52{color:#990000;font-size:36pt;font-family:"Philosopher";font-style:normal}.c58{font-size:12pt;font-family:"Pontano Sans";color:#000000;font-weight:400}.c16{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:justify}.c43{padding-top:8pt;padding-bottom:0pt;line-height:1.5;text-align:justify}.c3{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c76{padding-top:14pt;padding-bottom:4pt;line-height:1.5;text-align:justify}.c79{padding-top:24pt;padding-bottom:6pt;line-height:1.5;text-align:justify}.c84{padding-top:18pt;padding-bottom:4pt;line-height:1.5;text-align:justify}.c0{padding-top:0pt;padding-bottom:0pt;line-height:1.5;text-align:justify}.c49{color:#434343;font-size:14pt;font-family:"Philosopher";font-style:normal}.c66{padding-top:0pt;padding-bottom:0pt;line-height:1.5;text-align:left}.c27{background-color:#ffff00;color:#ff0000;text-decoration:none}.c21{font-family:"Philosopher";color:#cc0000;font-weight:400}.c39{font-family:"Comic Sans MS";color:#cc0000;font-weight:400}.c73{-webkit-text-decoration-skip:none;text-decoration:underline;text-decoration-skip-ink:none}.c56{font-weight:400;font-family:"Inconsolata";font-style:normal}.c70{color:#666666;font-weight:400;font-family:"Pontano Sans"}.c57{font-size:9pt;font-family:"Inconsolata";font-weight:400}.c54{font-size:20pt;font-family:"Arial";font-style:normal}.c87{background-color:#ffffff;max-width:451.3pt;padding:72pt 72pt 72pt 72pt}.c51{background-color:#ffffff;color:#222222;font-size:9pt}.c15{color:inherit;text-decoration:inherit}.c75{font-size:24pt;font-family:"Philosopher"}.c81{background-color:#ffffff;color:#666666}.c32{font-weight:400;font-family:"Consolas"}.c30{padding:0;margin:0}.c78{font-size:12pt;font-family:"Pontano Sans"}.c9{color:#000000;text-decoration:none}.c64{font-weight:400;font-family:"Inconsolata"}.c67{background-color:#ffffff;color:#222222}.c17{font-size:9pt;font-weight:700}.c25{margin-left:36pt}.c63{color:#999999}.c38{font-weight:700}.c23{height:12pt}.c83{font-family:"Pontano Sans"}.c80{font-size:24pt}.c55{font-family:"Consolas"}.c33{text-indent:36pt}.c82{background-color:#ffff00}.c40{text-decoration:none}.c48{vertical-align:baseline}.c37{height:0pt}.c61{font-size:9pt}.c68{font-size:8pt}.c4{margin-left:54pt}.c28{padding-left:0pt}.c45{font-size:11pt}.c74{height:18pt}.c85{background-color:#fff2cc}.c59{font-style:normal}.c72{background-color:#d9d9d9}.c7{font-style:italic}.c86{font-size:12pt}.c2{color:#980000}.c46{font-size:10pt}.c31{margin-left:18pt}.title{padding-top:0pt;color:#990000;font-weight:700;font-size:36pt;padding-bottom:0pt;font-family:"Philosopher";line-height:1.5;text-align:justify}.subtitle{padding-top:0pt;color:#666666;font-size:13pt;padding-bottom:10pt;font-family:"Trebuchet MS";line-height:1.5;font-style:italic;text-align:justify}li{color:#000000;font-size:12pt;font-family:"Pontano Sans"}p{margin:0;color:#000000;font-size:12pt;font-family:"Pontano Sans"}h1{padding-top:10pt;color:#666666;font-size:24pt;padding-bottom:0pt;font-family:"Philosopher";line-height:1.5;text-align:justify}h2{padding-top:10pt;color:#000000;font-size:18pt;padding-bottom:0pt;font-family:"Philosopher";line-height:1.5;text-align:justify}h3{padding-top:8pt;color:#434343;font-weight:700;font-size:14pt;padding-bottom:0pt;font-family:"Philosopher";line-height:1.5;text-align:justify}h4{padding-top:8pt;-webkit-text-decoration-skip:none;color:#666666;text-decoration:underline;text-decoration-skip-ink:none;font-size:11pt;padding-bottom:0pt;font-family:"Trebuchet MS";line-height:1.5;text-align:justify}h5{padding-top:8pt;color:#666666;font-size:11pt;padding-bottom:0pt;font-family:"Trebuchet MS";line-height:1.5;text-align:justify}h6{padding-top:8pt;color:#666666;font-size:11pt;padding-bottom:0pt;font-family:"Trebuchet MS";line-height:1.5;font-style:italic;text-align:justify}</style></head><body class="c87 doc-content"><p class="c0 c23"><span class="c54 c9 c38 c48"></span></p><p class="c0 c23"><span class="c54 c9 c38 c48"></span></p><p class="c0 c23"><span class="c9 c38 c48 c54"></span></p><p class="c24 title" id="h.mtuhyunovy7w"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 177.00px; height: 111.00px;"><img alt="" src="images/image14.png" style="width: 177.00px; height: 111.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c24 title" id="h.2fpr2lcfatly"><span class="c21">Scaffolding with </span><span class="c21">ScaMPI<br></span><span class="c63 c80">User Manual</span></p><p class="c66 c23"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c18"><span class="c8 c9"></span></p><p class="c66 c23"><span class="c8 c9"></span></p><p class="c66 c23"><span class="c8 c9"></span></p><p class="c66 c23"><span class="c8 c9"></span></p><p class="c60"><span class="c41 c9">ScaMPI - 2010, CRIBI - University of Padua</span></p><p class="c60"><span class="c3 c46"><a class="c15" href="https://www.google.com/url?q=http://genomics.cribi.unipd.it/scampi/&amp;sa=D&amp;source=editors&amp;ust=1666000537198060&amp;usg=AOvVaw3aKvYcjnEK0EhF2CJFL1q4">http://genomics.cribi.unipd.it/scampi/</a></span><span class="c41 c9">&nbsp;</span></p><p class="c60"><span class="c46">Andrea Telatin - &nbsp;</span><span class="c3 c46"><a class="c15" href="mailto:andrea.telatin@cribi.unipd.it">andrea.telatin@cribi.unipd.it</a></span><span class="c41 c9">&nbsp;</span></p><p class="c60"><span class="c41 c9">Version 1.16 - Last revision 29/08/2013</span></p><hr style="page-break-before:always;display:none;"><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><a id="id.8zyxtvo9yqd0"></a><p class="c0 title" id="h.jcb3ocf1qsn"><span class="c52 c38 c40 c48">Table of Contents</span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.g0o5viosveni">Preliminary operations and concepts</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.8ey2d2o3ixqz">What ScaMPI is</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.403pu78w3xev">Website</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.2nmckdkybe46">Nomenclature used in the manual</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.sei5egmcoxa6">System requirements</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.agu11qj33pih">Input files</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.hx9n7gflp8h">Preparing database</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.wiyca5wemwso">Contigs correctness check</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.26x6wbn1k3g">Populating the database</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.i6hhvzkt5cmm">Preparing contigs table</a></span></p><p class="c0 c4"><span class="c3"><a class="c15" href="#h.7hnt6we77pj7">NOTE: Many users like to deal with simple (short) contigs name. The best choice to do this is to rename the contigs FASTA file prior to populate the database AND performing the alignments.</a></span></p><p class="c0 c4"><span class="c3"><a class="c15" href="#h.7n0ldh64afwb">Other preliminary operations for the &ldquo;contigs.fa&rdquo; file</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.d7t473klss2o">Alignments of paired reads against contigs</a></span></p><p class="c0 c4"><span class="c3"><a class="c15" href="#h.bgdvqm7dfz1s">Making alignments</a></span></p><p class="c0 c4"><span class="c3"><a class="c15" href="#h.xup37bbyqha5">Pairing alignments</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.z0m3zq4av1i4">Preparing &ldquo;arcs&rdquo; table</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.3z1qd22amj2h">Launching the web interface</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.bcfe4misdln1">Requirements</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.k2w36nes5fl">Configuring the web directory</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.sdvvus6wkdkp">Automatic scaffolding from seed</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.lnqikkxb7jmj">Whole genome scaffolding</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.s3e8f7duoo7z">Scaffolding via web-interface</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.tsd66xs93650">Scaffolding via command line</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.duft8dwglgp7">Validating scaffolds with BAC ends</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.iiog2b8zahul">Convert chromatograms to FASTA files</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.bma29q4o2vk4">Aligning BAC ends</a></span></p><p class="c0 c4"><span class="c3"><a class="c15" href="#h.t9wjha73xkb4">Possible scaffold joining</a></span></p><p class="c0 c4"><span class="c3"><a class="c15" href="#h.wxc96xdwdozs">Confirmed scaffolds</a></span></p><p class="c0 c4"><span class="c3"><a class="c15" href="#h.i3uewm1jt28e">Summary log</a></span></p><p class="c0 c4"><span class="c3"><a class="c15" href="#h.tesarwftwyz4">Graphical Output</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.cgmlf9vw9d0v">Manual refinements</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.wdnpmskf72hx">Try exploring MP graph</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.73f4uoljz4uz">Join scaffolds</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.b2bmvmcf2p29">Appendix I &ndash; Core scripts</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.1qo7lg6044t6">Appendix II &ndash; Paired alignments</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.7finujinv245">Aligning and pairing SOLiD Mate Paired reads</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.imol8vsh039">Appendix III &ndash; Contig correctness check</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.jp00ill4hipa">Appendix IV - Telomere identification</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.twlbboe8ouo2">TRAP overview</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.4zvtntx2qnm9">De novo identification of the repeat</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.lf5orxq9tzze">Isolation of motif-containing MP reads</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.jsn1n3mzj9cd">Alignment of selected reads</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.sr4hf46m73j">Extraction of putative peri-telomeric regions</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.80sg42ok1843">Appendix V - Default Plug-ins</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.xxhaxxlkd773">Installing new plug ins</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.s9zfsrysg86k">Backup DB</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.2x7axij1wj9c">BLAST</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.g5sqcot0aaq6">Make AGP</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.165ipncd9vox">Appendix VI - File formats</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.m3jdp1ms1o5m">Input files</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.2zbsvz8abfwc">Alignments of MP/PE: format and pairing</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.v64zhypitvii">Arcs table</a></span></p><p class="c0 c31"><span class="c3"><a class="c15" href="#h.lk49h9d6n9lh">Appendix VII - Supplementary tools</a></span></p><p class="c0 c25"><span class="c3"><a class="c15" href="#h.4u1mttywcfhx">Graphical plot of scaffolds (GraphViz)</a></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p><hr style="page-break-before:always;display:none;"><h1 class="c12" id="h.i0wej4al4tsn"><span class="c6"></span></h1><h1 class="c20" id="h.g0o5viosveni"><span class="c6">Preliminary operations and concepts</span></h1><h2 class="c20" id="h.8ey2d2o3ixqz"><span class="c22 c9">What ScaMPI is</span></h2><p class="c0"><span class="c8 c9">ScaMPI is a suite of tools for genome scaffolding providing a visual interface. ScaMPI allows to:</span></p><ul class="c30 lst-kix_6aptd0j42w1q-0 start"><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Check the correctness of contigs</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Perform automatic scaffolding</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Visualize/browse connections via the web interface</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Validate the scaffold with a BAC ends validation program</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Identify telomeres with the TRAP program</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Perform BLASTs, ORF finding and primer picking via the web interface.</span></li></ul><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>ScaMPI is </span><span class="c38">modular</span><span class="c8 c9">&nbsp;meaning that while it provides all the tools needed for the above operations, the user can:</span></p><ul class="c30 lst-kix_rxkpu1ydcip2-0 start"><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">use its favourite aligner (SAM output required)</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">upload its own scaffolds (in AGP format)</span></li></ul><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">ScaMPI requires a little knowledge of Linux to be installed, while the aspects that could be relevant for the non bioinformatics aware collaborators are available to a web based interface that allows them to:</span></p><ul class="c30 lst-kix_6c526n8mx7cb-0 start"><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Browse contigs and their connections</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Perform BLASTs against contigs</span></li></ul><p class="c0"><span>These two features are useful to make data available for research before the final assembly is released. Moreover, being web based, once that the program is installed in a single PC, the interface is available without installation.</span></p><h2 class="c20" id="h.403pu78w3xev"><span class="c22 c9">Website</span></h2><p class="c0"><span>The official website for ScaMPI is </span><span class="c3"><a class="c15" href="https://www.google.com/url?q=http://genomics.cribi.unipd.it/scampi/&amp;sa=D&amp;source=editors&amp;ust=1666000537209590&amp;usg=AOvVaw0lZcdYhjdjSaUlalk9c5gH">http://genomics.cribi.unipd.it/scampi/</a></span><span>, that features a </span><span class="c3 c38"><a class="c15" href="https://www.google.com/url?q=http://genomics.cribi.unipd.it/scampi/news/&amp;sa=D&amp;source=editors&amp;ust=1666000537209880&amp;usg=AOvVaw2E8LdkcE6Tdag0sHTHWT0c">blog</a></span><span class="c8 c9">&nbsp;that is updated with tutorials, new tools and packages.</span></p><p class="c0"><span>We recommend to check the online blog for minor updates.</span></p><h2 class="c20" id="h.2nmckdkybe46"><span class="c22 c9">Nomenclature used in the manual</span></h2><p class="c0"><span class="c8 c9">Before starting let&rsquo;s consider a simple example of three contigs connected by SOLiD mate paired reads. We will refer to connection between contigs using the 3&rsquo;/5&rsquo; nomenclature of nucleic acids but omitting the &ldquo;prime&rdquo;, thus the link between contig1 and contig2 is &ldquo;3-3&rdquo; while connection between contig2 and contig3 is &ldquo;5-3&rdquo;.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 620.00px; height: 227.27px;"><img alt="" src="images/image16.png" style="width: 620.00px; height: 227.27px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c17">Figure 1</span><span class="c61">&nbsp;&ndash;</span><span class="c29 c9">&nbsp;Mate Paired reads aligned against reference contigs are used to produce &ldquo;arcs&rdquo;, later exploited for scaffolding.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>We call &ldquo;</span><span class="c38">arcs</span><span class="c8 c9">&rdquo; the connections between contigs. Each arc has:</span></p><ul class="c30 lst-kix_nmlk7fvtpy78-0 start"><li class="c0 c25 c28 li-bullet-0"><span>a </span><span class="c38">direction</span><span class="c8 c9">&nbsp;expressed as the two extremities of joined contigs (as 3-3, 3-5, etc)</span></li><li class="c0 c25 c28 li-bullet-0"><span>a </span><span class="c38">number of pairs/mates</span><span class="c8 c9">&nbsp;composing it (6 in the contig1-contig2 example)</span></li><li class="c0 c25 c28 li-bullet-0"><span>a </span><span class="c38">concordance percentage</span><span class="c8 c9">, representing the fraction of pairs confirming the direction (83% in the first example, because 5 out of 6 pairs confirm the 3-3 direction.</span></li><li class="c0 c25 c28 li-bullet-0"><span>an </span><span class="c38">average insert</span><span>, that is the average size &ndash; </span><span class="c7">gap excluded &ndash;</span><span>&nbsp;of the region between the two mate/pairs. Ideally the average library size minus the average insert gives an estimate of the gap size.</span></li></ul><hr style="page-break-before:always;display:none;"><h2 class="c20 c74" id="h.8qu9uk3g9436"><span class="c22 c9"></span></h2><h2 class="c20" id="h.sei5egmcoxa6"><span class="c22 c9">System requirements</span></h2><p class="c0"><span class="c8 c9">For ScaMPI to work you&rsquo;ll need a *AMP machine (Apache/MySQL/PHP) with Perl installed. A 64-bit Linux distribution with MySQL client/server available is the ideal and the only actively supported platform, although with minimum efforts it is possible to run both the scripts and the web interface under OS X and Windows. The whole ScaMPI pipeline has been tested on Ubuntu 8.04 LTS, 12.04 LTS, Linux Mint 12 and CentOS 6. </span></p><p class="c0"><span>As a proof of concept the whole scaffolding process, and the web interface, has been tested on a Raspberry Pi (ARM - 512 Mb RAM, LAMP installed), and worked fine. Most </span><span class="c38">browsers</span><span class="c8 c9">&nbsp;will work fine with the interface, although Google Chrome&trade; is recommended.</span></p><p class="c0"><span>RAM is an issue when </span><span class="c38">aligning</span><span>, and the size required increases with the size of the reference (</span><span class="c7">i.e.</span><span class="c8 c9">&nbsp;sum of contigs). Nevertheless, the alignment step can be performed on a separate server.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>ScaMPI core scripts</span><span class="c63">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;. . . . . . . . . . . .&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c8 c9">Perl and MySQL (tested on GNU/Linux)</span></p><p class="c0"><span>ScaMPI web gui</span><span class="c63">&nbsp;. . . . . . . . . . . . . . . . .</span><span class="c8 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Apache (or other web server), MySQL, PHP</span></p><p class="c0"><span>PASS aligner </span><span class="c63">. . . . . . . . . . . . . . . . . . . .&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c8 c9">64-bit GNU/Linux</span></p><h2 class="c20" id="h.agu11qj33pih"><span class="c22 c9">Input files</span></h2><p class="c0"><span class="c8 c9">It is handy to create a directory with symbolic links to the following files (or with the following files themselves):</span></p><p class="c0 c25"><span>contigs.fasta </span><span class="c63">. . . . . . . . . . . .&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c8 c9">Set of contigs to be scaffolded (multifasta)</span></p><p class="c0 c25"><span>lib1_F3.csfastq </span><span class="c63">. . . . . . . . . . &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c8 c9">First mate/pair for &ldquo;library1&rdquo;</span></p><p class="c0 c25"><span>lib1_R3.csfastq </span><span class="c63">. . . . . . . . . . &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c8 c9">Second mate/pair for &ldquo;library1&rdquo;</span></p><p class="c0 c25"><span class="c8 c9">&hellip;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add as many libraries is possible</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>SOLiD libraries are expected to be in &ldquo;csfastq&rdquo; format (you can combine the .csfasta and the .qual files with the </span><span class="c2">csfasta_to_fastq</span><span class="c8 c9">&nbsp;program in the &ldquo;tools&rdquo; directory).</span></p><a id="id.4axn9x682ji"></a><h2 class="c20" id="h.hx9n7gflp8h"><span class="c22 c9">Preparing database</span></h2><p class="c0"><span class="c38">Automatic database creation</span><span>&nbsp;- The </span><span class="c2">scampi_project.pl</span><span class="c8 c9">&nbsp;script allows for automatic creation of a database and configuration file for the ScaMPI program and web interface. </span></p><p class="c0"><span>If launched with the </span><span class="c7">-c </span><span class="c8 c9">switch will create both the database and the configuration file, otherwise will produce the configuration file alone assuming that the database is already present.</span></p><p class="c16"><span class="c64 c46">&nbsp;+</span><span class="c64 c46">---------------------------------------------------------------------+<br> | ScaMPI TOOLS &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;|<br> | Create project files &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;|<br> +---------------------------------------------------------------------+<br><br> Will save a &#39;db_data.php&#39; file to be used in other scripts.<br> <br> -p &nbsp; &nbsp; Project name (short string like &#39;wheat&#39;) <br> -h &nbsp; &nbsp; MySQL host [localhost]<br> -u &nbsp; &nbsp; Desired MySQL user name [scampi]<br> -p &nbsp; &nbsp; Desired MySQL password [magic]<br> -d &nbsp; &nbsp; Desired MySQL database [equal to -p]<br> -o &nbsp; &nbsp; Output directory [.]<br> -c &nbsp; &nbsp; Create database and user (requires administrative password)</span><span class="c57"><br></span></p><p class="c0"><span>The </span><span class="c2">scampi_testconfig.pl</span><span class="c8 c9">&nbsp;script can be used to test the configuration file and check that the connection to the database works.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c38">Manual database preparation </span><span>&ndash; Create a MySQL database with your project&rsquo;s name. It could be a good idea to create a dedicate user with privileges on that database. We will refer to these data as </span><span class="c7">dbuser, dbpwd</span><span>&nbsp;and </span><span class="c7">dbname</span><span class="c8 c9">.</span></p><p class="c0"><span class="c8 c9">Example:</span></p><p class="c0"><span class="c61">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c32 c61">mysql -h </span><span class="c32 c61 c7">host </span><span class="c32 c61">-u </span><span class="c32 c7 c61">root</span><span class="c32 c61">&nbsp;-p </span><span class="c32 c9 c48 c61 c7">password</span></p><p class="c0 c33"><span class="c40 c48 c61 c7 c70">inside the MySQL shell</span></p><p class="c0"><span class="c32 c61">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&gt; CREATE DATABASE </span><span class="c32 c61 c7">dbname</span><span class="c14 c9">;</span></p><p class="c0 c25"><span class="c51 c32">&gt; CREATE USER &#39;</span><span class="c51 c32 c7">dbuser</span><span class="c51 c32">&#39;@&#39;localhost&#39; IDENTIFIED BY &#39;</span><span class="c32 c7 c51">dbpwd</span><span class="c51 c32">&#39;;<br>&gt; GRANT ALL ON </span><span class="c32 c61 c7">dbname</span><span class="c51 c32">.* TO &#39;</span><span class="c51 c32 c7">dbuser</span><span class="c14 c40 c67">&#39;@&#39;localhost&#39;;</span></p><p class="c0 c23"><span class="c14 c67 c40"></span></p><h2 class="c20" id="h.wiyca5wemwso"><span class="c22 c9">Contigs correctness check</span></h2><p class="c0"><span class="c8 c9">Contigs assembled using non paired reads could lead to misassemblies that paired reads can solve. Before starting the scaffolding step it&rsquo;s important to have the most robust contigs dataset as possible.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>See </span><span class="c3"><a class="c15" href="#id.93kod45arhxj">Appendix III</a></span><span class="c8 c9">&nbsp;to see how to use the provided correctness check module.</span></p><h1 class="c20" id="h.26x6wbn1k3g"><span class="c6">Populating the database</span></h1><a id="id.ox73powlhv51"></a><h2 class="c20" id="h.i6hhvzkt5cmm"><span class="c9 c22">Preparing contigs table</span></h2><p class="c0"><span class="c58 c40 c48 c7">ScaMPI retrieves information about contigs (name, size, coverage) from a table of the database that can be created starting from the contigs file. </span></p><p class="c0"><span>The program </span><span class="c2">contigs_to_db.pl</span><span>&nbsp;requires a multifasta file containing contigs and will extract contig names and length from the same. If the header has the coverage information (like some assembler does) it can be used to save this important feature in the table. The script produce as output a SQL script to create a &ldquo;contigs&rdquo; table.</span></p><p class="c0"><span>A basic usage of the </span><span class="c8 c9">program:</span></p><p class="c0"><span class="c19 c9">&nbsp; contigs_to_db.pl contigs.fasta &gt; contigs_table.sql</span></p><p class="c0"><span class="c32">&nbsp; mysql -h </span><span class="c32 c7">host </span><span class="c32">-u </span><span class="c32 c7">dbuser</span><span class="c32">&nbsp;-p </span><span class="c32 c7">dbpwd</span><span class="c32">&nbsp;-D </span><span class="c32 c7">dbname</span><span class="c19 c9">&nbsp;&lt; contigs_table.sql</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>If your contigs headers contains information about </span><span class="c38">average coverage</span><span class="c8 c9">, this can be (and should be) saved to the database. </span></p><p class="c0"><span class="c8 c9">If the header is in the Abyss format it will look like:</span></p><p class="c0 c33"><span class="c19 c9">&gt;contig1|size1517|read89|cov5.00|seed:1</span></p><p class="c0 c23 c33"><span class="c19 c9"></span></p><p class="c0"><span>You can use the</span><span class="c7">&nbsp;</span><span class="c32 c7">-covstring </span><span class="c32">&quot;cov#\|&quot;</span><span>&nbsp;switch to extract the information. The </span><span class="c7">-covstring</span><span>&nbsp;parameter expects a pattern where the # character is replaced with numbers and dots. This parameter can be used with any kind of assembler that prints directly the coverage in contigs headers, it&rsquo;s important to escape slashes and pipes. For Velvet you can use a </span><span class="c32 c7">-covstring</span><span class="c19 c9">&nbsp;&ldquo;cov_#&rdquo;;</span></p><p class="c0"><span>If you use </span><span class="c38">Newbler</span><span>&nbsp;you&rsquo;ll have the number of reads used; the script provides the </span><span class="c7">-newbler&nbsp;AverageReadLength</span><span class="c8 c9">&nbsp;parameter to estimate the average coverage.</span></p><p class="c0"><span class="c8 c9">With Newbler contigs:</span></p><p class="c0"><span class="c39">&nbsp; </span><span class="c19 c9">&nbsp;contigs_to_db.pl contigs.fasta -newbler 390 &gt; contigs.sql</span></p><h3 class="c43" id="h.7hnt6we77pj7"><span class="c5">NOTE</span><span class="c58 c85">:</span><span class="c58">&nbsp;Many users like to deal with simple (short) contigs name. The best choice to do this is to rename the contigs FASTA file prior to populate the database AND performing the alignments.</span></h3><h3 class="c43" id="h.7n0ldh64afwb"><span>Other preliminary operations for the &ldquo;contigs.fa&rdquo; file</span></h3><p class="c0"><span class="c8 c9">The &ldquo;contigs.fa&rdquo; file is a pivotal resource in ScaMPI and can be used not only to align mate paired information but also for other optional steps. In particular:</span></p><ul class="c30 lst-kix_ur78f3m641k9-0 start"><li class="c0 c25 c28 li-bullet-0"><span class="c38">BLAST</span><span class="c8 c9">&nbsp;- The web interface has a plug in for BLASTing against contigs. </span></li></ul><p class="c0 c25"><span>See </span><span class="c3"><a class="c15" href="#id.it6whu3pmxr3">Appendix V</a></span><span class="c8 c9">.</span></p><ul class="c30 lst-kix_ur78f3m641k9-0"><li class="c0 c25 c28 li-bullet-0"><span class="c38">Other tools</span><span>&nbsp;- Some tools requires to find a single file per contig in the </span><span class="c32">/web/contigs</span><span>&nbsp;directory. To produce it use the </span><span class="c2">split_contigs.pl</span><span>&nbsp;program.</span></li></ul><a id="id.c5n6x4mdk37q"></a><h2 class="c20" id="h.d7t473klss2o"><span class="c22 c9">Alignments of paired reads against contigs</span></h2><p class="c0"><span class="c58 c40 c48 c7">SOLiD Mate Paired are better aligned with the PASS program, that is supplied with ScaMPI. A main advantage of PASS is a robust implementation of color space reads, using SOLiD mate pairs, and the platform independent output produced with the pairing program.</span></p><h3 class="c43" id="h.bgdvqm7dfz1s"><span class="c49 c38 c40 c48">Making alignments</span></h3><p class="c0"><span>If using PASS v.1.64 align separately the FOR and REV pairs. A PASS release tested to be used with ScaMPI is stored in the tools directory (compiled for Linux 64bit). The binary is called </span><span class="c2">pass</span><span class="c8 c9">.</span></p><p class="c0"><span>Note that </span><span class="c38">any aligner</span><span>&nbsp;can be used for ScaMPI, as long as the two reads set (For and Rev) are aligned independently. The output has to be converted from SAM to GFF with the included </span><span class="c2">sam2gff</span><span class="c8 c9">&nbsp;utility.</span></p><p class="c0"><span class="c8 c9">For SOLiD reads use the following command, for Illumina just replace &ldquo;-csfastq&rdquo; with &ldquo;-fastq&rdquo; followed by the FASTQ file of the reads:</span></p><p class="c0"><span class="c9 c19">&nbsp; pass -csfastq lib1_F3.csfastq -d contigs.fasta \</span></p><p class="c0"><span class="c32">&nbsp; -cpu </span><span class="c32 c7">n</span><span class="c19 c9">&nbsp;-gff -fid 90 -uniq &gt; lib1_F3.gff 2&gt; log_lib1_F3</span></p><a id="id.rc7ixubsbjat"></a><h3 class="c76" id="h.xup37bbyqha5"><span class="c49 c38 c40 c48">Pairing alignments</span></h3><p class="c0"><span>The pairing process checks the alignment for both mates (or pairs) and classify them into classes. We&rsquo;ll use the </span><span class="c2">pass_pair</span><span class="c8 c9">&nbsp;program supplied in the &ldquo;tools&rdquo; directory.</span></p><p class="c0"><span class="c8 c9">For scaffolding purposes when one mate is uniquely aligned against a contig and the other in another contig (uniquely again), this is classified as UNIQUE_PAIR_OUT (provided that the distance between the two is compatible with insert size).</span></p><p class="c0"><span>When both mates align uniquely within the same contig with the proper orientation and distance they are classified as UNIQUE_PAIR. Please, create a new directory to store output files (in the example is </span><span class="c7">outputdir</span><span class="c8 c9">/).</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c38">Reads directionality</span><span>&nbsp;&ndash; SOLiD Mate Paired reads (tagged as _</span><span class="c7">F3</span><span>&nbsp;and _</span><span class="c7">R3</span><span>) have the same direction (F3&rarr;, R3&rarr;) while paired-ends (for example Illumina) reads have opposite directions and are sometimes tagged as /</span><span class="c7">1</span><span>&nbsp;and /</span><span class="c7">2</span><span>&nbsp;(1&rarr;,&larr;2). The pass_pair program needs to know exactly this to sort out categories: </span><span class="c7">-pe_type 0</span><span class="c7">&nbsp;</span><span>specifies the SOLiD type, </span><span class="c7">-lib_type 0</span><span>&nbsp;for SOLiD direct libraries. For further details and for other seqeuencing platforms refer to PASS manual available at </span><span class="c3"><a class="c15" href="https://www.google.com/url?q=http://pass.cribi.unipd.it/&amp;sa=D&amp;source=editors&amp;ust=1666000537222757&amp;usg=AOvVaw3tDuDGu0e0xTRnFAh8Gg-Z">http://pass.cribi.unipd.it/</a></span><span class="c8 c9">.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c38">Library insert size</span><span>&nbsp;&ndash; </span><span class="c2">pass_pair</span><span>&nbsp;requires the user to provide a range for insert size plus a maximum size. If you have a library spanning 500-1000 bp you could provide 400 1100 1500 as parameters. The third is the maximum tolerated distance (that can be ignored for ScaMPI and set to </span><span class="c7">maximum</span><span>+1). Please, consider that the log file of </span><span class="c2">pass_pair</span><span class="c8 c9">&nbsp;provided a detailed distribution of insert sizes calculated on large contigs. A preliminary pairing with &ldquo;relaxed&rdquo; parameters could be used to fine tune this important setting.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">For SOLiD Mate Paired reads:</span></p><p class="c0"><span class="c26 c9">&nbsp; pass_pair -gff1 lib1_F3.gff -gff2 lib1_R3.gff -ref contigs.fa \</span></p><p class="c0"><span class="c26 c9">&nbsp; -range 400 1200 1500 -o outputdir/ -log outputdir/log.txt</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">For Illumina Paired Ends reads:</span></p><p class="c0"><span class="c26 c9">&nbsp; pass_pair -gff1 lib1_F3.gff -gff2 lib1_R3.gff -ref contigs.fa \</span></p><p class="c0"><span class="c26 c9">&nbsp; -range 500 1000 1500 -tags &#39;/1&#39; &#39;/2&#39; -o outputdir/ -log outputdir/log.txt</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">In the output directory the program will save many files. We are going to use:</span></p><ul class="c30 lst-kix_r5gz416xmm5m-0 start"><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">UNIQUE_PAIR&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Both mates are aligned inside the same contig</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">UNIQUE_PAIR_OUT&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The two mates align in different contigs</span></li></ul><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">In both cases they represent unique alignments with proper mutual orientation and insert size. The format is a GFF file, exactly as the input alignment files, with the only difference that each alignment line is followed by the alignment of the other mate.</span></p><a id="id.q75r1kyfsiqn"></a><h2 class="c84" id="h.z0m3zq4av1i4"><span class="c22 c9">Preparing &ldquo;arcs&rdquo; table</span></h2><p class="c0"><span class="c58 c40 c48 c7">The second pivotal table is the one for connections between contigs. A single table can store different libraries, as long as each library has its unique identifier.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c2">mates_to_arcs.pl</span><span>&nbsp;parses the (gff) UNIQUE_PAIR_OUT file produced by </span><span class="c2">pass_pair</span><span>&nbsp;and produces a tabular report of arcs. The list of parameters, that can invoked launching the program alone or with the</span><span class="c7">&nbsp;-h</span><span>&nbsp;switch, is shown below:</span></p><p class="c16 c23"><span class="c8 c9"></span></p><p class="c16"><span class="c64 c68">&nbsp; &nbsp; &nbsp;___ &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;__ &nbsp;__ &nbsp;___ &nbsp;___ <br></span><span class="c64 c46">&nbsp; &nbsp;/ __| __ __ _| &nbsp;\/ &nbsp;| _ \_ _|<br> &nbsp; \__ \/ _/ _` | |\/| | &nbsp;_/| | <br> &nbsp; |___/\__\__,_|_| &nbsp;|_|_| |___|<br> &nbsp;-------------------------------------------------------------------------------<br> &nbsp;MATE PAIRS (UNIQUE_PAIR_OUT) to ARCS &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;CRIBI Biotech Center 2012<br> &nbsp;-------------------------------------------------------------------------------<br><br> &nbsp;This program parses the UNIQUE_PAIR_OUT output of &#39;pass_pair&#39; to create &#39;arcs&#39;<br> &nbsp;for scaffolding. Default output is a tabular file as described in the manual.<br> &nbsp; <br> &nbsp; &nbsp;-i or -f &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;Input file (UNIQUE_PAIR_OUT from &#39;pass_pair&#39; (GFF)<br> &nbsp; &nbsp;-m, --minarcs &nbsp; &nbsp; Minimum number of mates to create a connection [3]<br> &nbsp; &nbsp;-c, --concordance Minimum % of concordant mates per connection [0]<br> &nbsp; &nbsp;-l &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;Maximum length of the connection [off]<br> &nbsp; &nbsp;-t, --test &nbsp; &nbsp; &nbsp; &nbsp;Stop after parsing this number of lines [off]<br> &nbsp; &nbsp;-d, --debug &nbsp; &nbsp; &nbsp; Print a debug column [off]<br><br> &nbsp; &nbsp;To print output in SQL format:<br> &nbsp; &nbsp;-tab &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;MySQL table name (eg: &#39;arcs&#39;)<br> &nbsp; &nbsp;-lib &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;Library id for the MySQL database (eg: &#39;1&#39;)<br> &nbsp; &nbsp;-head &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Print also a create table statement <br> &nbsp; &nbsp;<br> &nbsp; &nbsp;-h, --help &nbsp; &nbsp; &nbsp; &nbsp;Prints this message<br> &nbsp; &nbsp;<br> &nbsp; &nbsp;Example:<br> &nbsp; &nbsp;scampi_mates_to_arcs.pl -i UNIQUE_PAIR_OUT -c 80 -m 10 &gt; library1.arcs<br> &nbsp;-------------------------------------------------------------------------------</span><span class="c8 c9"><br></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">This program clusters alignments sharing the same origin and destination contig, and will print the most abundant direction (concordance percentage). With the -c and -m switches we can impose to print only arcs with a minimum concordance percentage and number of mates composing it. The default output format is tabular:</span></p><p class="c0 c23"><span class="c8 c9"></span></p><a id="t.ccf8d9e8cc5c53cebaeb4a56909a5bb083d1e49b"></a><a id="t.0"></a><table class="c53"><tr class="c37"><td class="c50 c72" colspan="1" rowspan="1"><p class="c16"><span>Ctg1 </span><span class="c58 c40 c48 c7">(from)</span></p></td><td class="c50 c72" colspan="1" rowspan="1"><p class="c16"><span>Ctg2</span><span class="c58 c40 c48 c7">&nbsp;(to)</span></p></td><td class="c62" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Ctg1_end</span></p></td><td class="c35 c72" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Ctg2_end</span></p></td><td class="c36 c72" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Distance</span></p></td><td class="c69 c72" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Concord.</span></p></td><td class="c47 c72" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">#Mates</span></p></td></tr><tr class="c37"><td class="c50" colspan="1" rowspan="1"><p class="c16"><span>contig040</span></p></td><td class="c50" colspan="1" rowspan="1"><p class="c16"><span>contig057</span></p></td><td class="c77" colspan="1" rowspan="1"><p class="c16"><span>3</span></p></td><td class="c35" colspan="1" rowspan="1"><p class="c16"><span>5</span></p></td><td class="c36" colspan="1" rowspan="1"><p class="c65"><span>900</span></p></td><td class="c69" colspan="1" rowspan="1"><p class="c65"><span>100</span></p></td><td class="c47" colspan="1" rowspan="1"><p class="c16"><span>1294</span></p></td></tr><tr class="c37"><td class="c50" colspan="1" rowspan="1"><p class="c16"><span>contig060</span></p></td><td class="c50" colspan="1" rowspan="1"><p class="c16"><span>contig079</span></p></td><td class="c77" colspan="1" rowspan="1"><p class="c16"><span>5</span></p></td><td class="c35" colspan="1" rowspan="1"><p class="c16"><span>5</span></p></td><td class="c36" colspan="1" rowspan="1"><p class="c65"><span>252</span></p></td><td class="c69" colspan="1" rowspan="1"><p class="c65"><span>99</span></p></td><td class="c47" colspan="1" rowspan="1"><p class="c16"><span>15</span></p></td></tr></table><p class="c0"><span class="c8 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p><p class="c0"><span>The </span><span class="c2">scampi_arcs2sql.pl</span><span>&nbsp;tool convert this tabular output into a SQL statement to create and populate the MySQL table. The latest version of mates_to_arcs.pl allows to print directly in SQL format, passing the -tab and -lib parameters that specify, respectively, the name of the table (</span><span class="c7">e.g.</span><span>&nbsp;arcs) and the name of the library (</span><span class="c7">e.g. </span><span class="c8 c9">1). The former should be arcs if working with just a dataset, the latter an integer either progressive or similar to the insert size.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><hr style="page-break-before:always;display:none;"><p class="c0 c23"><span class="c8 c9"></span></p><h1 class="c20" id="h.3z1qd22amj2h"><span class="c6">Launching the web interface</span></h1><h2 class="c20" id="h.bcfe4misdln1"><span class="c22 c9">Requirements</span></h2><ol class="c30 lst-kix_odlzlhcdt35z-0 start" start="1"><li class="c0 c25 c28 li-bullet-0"><span>You should have MySQL credentials (host, username, password and database name) as </span><span class="c3"><a class="c15" href="#id.4axn9x682ji">described before</a></span><span class="c8 c9">.</span></li><li class="c0 c25 c28 li-bullet-0"><span>You should have populated the </span><span class="c7">contigs</span><span>&nbsp;table (see </span><span class="c3"><a class="c15" href="#id.ox73powlhv51">Preparing the contigs table</a></span><span class="c8 c9">).</span></li><li class="c0 c25 c28 li-bullet-0"><span>You should have created &ldquo;arcs&rdquo; from Mate Paired alignments (see </span><span class="c3"><a class="c15" href="#id.c5n6x4mdk37q">Alignments of Mate Pairs</a></span><span>) and populated the </span><span class="c7">arcs</span><span>&nbsp;table (see P</span><span class="c3"><a class="c15" href="#id.q75r1kyfsiqn">reparing the arcs table</a></span><span class="c8 c9">)</span></li></ol><h2 class="c20" id="h.k2w36nes5fl"><span class="c22 c9">Configuring the web directory</span></h2><p class="c0"><span>The ScaMPI package contains a &ldquo;web&rdquo; directory that has to be accessible via web, either configuring the web server or just placing it inside the www documents directory (e.g. in</span><span class="c64">&nbsp;/var/www/</span><span class="c8 c9">&nbsp;for defaults Apache&rsquo;s installations).</span></p><p class="c0"><span>In the </span><span class="c64">/web/inc </span><span>subdirectory the </span><span class="c2">db_con.php</span><span class="c8 c9">&nbsp;file has to be changed inserting the MySQL credentials.</span></p><p class="c0"><span>Once that the db_con.php file is updated you can browse to the </span><span class="c32">/web</span><span>&nbsp;directory using a web browser. If you placed the whole </span><span class="c7">scampi</span><span>&nbsp;directory in the document root directory (e.g. </span><span class="c64">/var/www</span><span>/) you can go to </span><span class="c3"><a class="c15" href="https://www.google.com/url?q=http://localhost/scampi/web/&amp;sa=D&amp;source=editors&amp;ust=1666000537234887&amp;usg=AOvVaw1PryqQjU1bfRpeGlXrYnzc">http://localhost/scampi/web/</a></span><span>&nbsp;to open the web interface. You&rsquo;ll see something like </span><span class="c3"><a class="c15" href="#id.l6tpdkptdn7x">Figure 2</a></span><span class="c8 c9">. The top bar contains shortcuts for different sections, while the main page resumes the current scaffolding status: the progress bar will indicate 0% (we will change things in a while), and you&rsquo;ll see a list of the largest contigs not in scaffold.</span></p><p class="c0"><span>Clicking into a contig name will open the page depicting its connections (see </span><span class="c3"><a class="c15" href="#id.x4vea6rrphbm">Figure 3</a></span><span class="c8 c9">). This web interface is already functional to perform some manual scaffolding. This can be useful to understand your genomes&rsquo; peculiar problematic regions, as well as the reasonable coverage limits to avoid entering in repetitive regions.</span></p><a id="id.l6tpdkptdn7x"></a><p class="c0 c23"><span class="c8 c9"></span></p><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 545.00px; height: 368.00px;"><img alt="" src="images/image15.png" style="width: 545.00px; height: 368.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c17">Figure 2</span><span class="c29 c9">&nbsp;- Default view of the ScaMPI web interface after populating the database with contigs and arcs informations. If accessed for the very first time will propose automatic scaffolding.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><a id="id.x4vea6rrphbm"></a><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 520.00px; height: 366.00px;"><img alt="" src="images/image10.png" style="width: 520.00px; height: 366.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c17">Figure 3</span><span class="c9 c29">&nbsp;- An example of &ldquo;contig&rdquo; view. You can see a list of connections either from 3&rsquo; or 5&rsquo; at the bottom, and the most abundant depicted in the central scheme. If clicking the &ldquo;Scaffolder&rdquo; button from this page, will perform scaffolding from selected contig as seed.</span></p><h2 class="c20" id="h.sdvvus6wkdkp"><span class="c22 c9">Automatic scaffolding from seed</span></h2><p class="c0"><span>From each contig page (see </span><span class="c3"><a class="c15" href="#id.x4vea6rrphbm">Figure 3</a></span><span>) you can start the scaffolding from seed clicking the &ldquo;</span><span class="c7">Scaffolder</span><span class="c8 c9">&rdquo; link in the top bar. This can be a first step to work with your data. You can inspect what causes the scaffolding to end by clicking to the final contigs of each automatic scaffold and seeing if there is a lack of arcs (low MP coverage) or an excess of arcs (repetitive/complex region).</span></p><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 491.00px; height: 363.00px;"><img alt="" src="images/image12.png" style="width: 491.00px; height: 363.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c17">Figure 4 </span><span class="c29 c9">- Scaffolding from &ldquo;contig00001&rdquo;. You can fine-tune the minimum number of connections and maximum coverage.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>The script used by the interface to perform the scaffolding is the &ldquo;divorce.pl&rdquo; module under the </span><span class="c64">/web/inc</span><span class="c8 c9">&nbsp;directory. </span></p><h1 class="c12" id="h.kjk9j8x6ze3c"><span class="c6"></span></h1><hr style="page-break-before:always;display:none;"><h1 class="c12" id="h.5g2njvbuq19j"><span class="c6"></span></h1><h1 class="c20" id="h.lnqikkxb7jmj"><span class="c6">Whole genome scaffolding</span></h1><p class="c0"><span>The first step to perform the automatic scaffolding is to invoke the </span><span class="c2">multi-scampi.pl</span><span>&nbsp;script (again in the </span><span class="c64">/web/inc</span><span class="c8 c9">&nbsp;directory).</span></p><p class="c0"><span>Although for this task is usually performed via the command line, there is a button labeled &ldquo;</span><span class="c7">Scaffolder</span><span>&rdquo; in the top bar of the web interface that access the</span><span class="c13">&nbsp;very </span><span class="c8 c9">same tool. If you enter the web interface for the first time, a link in the home page will suggest you to try the automatic scaffolder (that can be tested without altering the database, first).</span></p><h2 class="c20" id="h.s3e8f7duoo7z"><span class="c22 c9">Scaffolding via web-interface</span></h2><p class="c0"><span class="c8 c9">Invoking &ldquo;Scaffolder&rdquo; from the home page allows to test ScaMPI algorithm on whole genome. This can be fine-tuned later, but it&rsquo;s usually a good starting point.</span></p><p class="c0"><span>Select a minimum number of arcs and the maximum coverage then click &ldquo;Scaffold&rdquo; button. Note that the process can take several minutes, but works on background so that you can even close the browser. There is a &ldquo;shortcut link&rdquo; on the bottom that can be used to return checking the scaffolding status later. Our testset (</span><span class="c7">N. gaditana</span><span class="c8 c9">, 30 Mb) took 4 minutes on a Pentium Core2 (Q9300) with 8Gb RAM.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 484.00px; height: 310.00px;"><img alt="" src="images/image7.png" style="width: 484.00px; height: 310.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c17">Figure 5 </span><span class="c29 c9">- The &ldquo;Scaffolder&rdquo; interface, if invoked from the homepage, is set to perform whole genome scaffolding. Selecting &ldquo;off&rdquo; in &ldquo;Write to database&rdquo; we simulate the process, otherwise scaffolds will be written after deleting previous entries.</span></p><p class="c0 c23"><span class="c29 c9"></span></p><p class="c0"><span class="c8 c9">You&rsquo;ll notice the process is finished - hopefully - by seeing that the progress bar in the home page, that counts how many bases are included in scaffolds (Figure 6).</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 522.00px; height: 461.00px;"><img alt="" src="images/image13.png" style="width: 522.00px; height: 461.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c38 c46">Figure 6</span><span class="c41 c9">&nbsp;- ScaMPI&rsquo;s home page now provides link to scaffolds and gives an idea of the amount of bases in scaffold thanks to the green progress bar.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><h2 class="c20" id="h.tsd66xs93650"><span class="c22 c9">Scaffolding via command line</span></h2><p class="c0"><span>The web interface is actually a front end for the &ldquo;</span><span class="c2">multi-scampi.pl</span><span>&rdquo; script in the </span><span class="c64">/web/inc </span><span class="c8 c9">directory.</span></p><p class="c0"><span>Here how it works:</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp; ___ &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;__ &nbsp;__ ___ ____</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp;/ __| __ __ _| &nbsp;\/ &nbsp;| _ \_ _|</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp;\__ \/ _/ _` | |\/| | &nbsp;_/| |</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp;|___/\__\__,_|_| &nbsp;|_|_| |___|</span></p><p class="c16"><span class="c10 c9">&nbsp; -------------------------------------------------------------------------------</span></p><p class="c16"><span class="c10 c9">&nbsp; SCAFFOLDING FROM SEEDS &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CRIBI Biotech Center 2010</span></p><p class="c16"><span class="c10 c9">&nbsp; -------------------------------------------------------------------------------</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-db &nbsp; FILE &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Configuration file for MySQL connection [./db_con.php]</span></p><p class="c16"><span class="c9 c10">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-div &nbsp;FILE &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scampi core module [./divorce.pl]</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-cov &nbsp;INT &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Maximum contig coverage [80]</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-arc &nbsp;INT &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Minimum amount of mates per arc [12]</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-con &nbsp;INT &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Minimum percentage of concordance [90]</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-fil &nbsp;FILE &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scampi optional module [], optional</span></p><p class="c16"><span class="c10 c9">&nbsp;</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-write &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Save scaffold in DB</span></p><p class="c16"><span class="c10 c9">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-rewrite &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Erase all previous info and save scaffold in DB</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>The script can be invoked without parameters as will use the </span><span class="c7">db_con.php </span><span>file inside its directory for database connection and other default settings, but for the first trial we recommend to use the</span><span class="c64">&nbsp;-fil NO</span><span>&nbsp;switch, that avoid using an optional module for adding smaller contigs skipped by first scaffolding. <br>Moreover in order to save to database the scaffold you have to use the</span><span class="c64">&nbsp;-rewrite</span><span>&nbsp;switch (</span><span class="c64">-write</span><span>&nbsp;if you want </span><span class="c38">not</span><span class="c8 c9">&nbsp;to clear previous scaffolds, that is not recommended in general).</span></p><p class="c0"><span class="c8 c9">After this first step you&rsquo;ll find that the progress bar of the main page will show the amount of bases included in scaffolds, and the list of scaffold will no longer be empty.</span></p><p class="c0"><span>A simple example of multi-scampi.pl execution (from a very limited hardware as a Raspberry Pi!) is shown in the video: </span><span class="c3"><a class="c15" href="https://www.google.com/url?q=https://vimeo.com/67809642&amp;sa=D&amp;source=editors&amp;ust=1666000537243411&amp;usg=AOvVaw17wNXdS0MzcJ16J0NOKkDq">https://vimeo.com/67809642</a></span><span class="c8 c9">. </span></p><hr style="page-break-before:always;display:none;"><p class="c0 c23"><span class="c8 c9"></span></p><h1 class="c20" id="h.duft8dwglgp7"><span class="c6">Validating scaffolds with BAC ends</span></h1><p class="c0"><span class="c8 c9">When possible, it is recommended to further experimental data to validate genomic scaffolds. A common procedure involves the sequencing of the two ends of large insert libraries as BAC ends or Fosmid ends.</span></p><p class="c0"><span class="c8 c9">We developed and tested a simple pipeline to simplify the analysis of the results.</span></p><p class="c0"><span class="c8 c9">It should be noted that a BAC ends pair properly aligned verifies the portion of scaffold physically covered by it. </span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 521.00px; height: 512.00px;"><img alt="" src="images/image5.png" style="width: 521.00px; height: 512.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c38 c46">Figure 7</span><span class="c41 c9">&nbsp;- Schematic representation of BAC ends validation cases</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">Other outcomes of the alignment, as depicted above, will give hints of:</span></p><ul class="c30 lst-kix_6meiqxvvvyw3-0 start"><li class="c0 c25 c28 li-bullet-0"><span class="c38">correct scaffolds </span><span class="c8 c9">- as stated above</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c38">superscaffolding</span><span>&nbsp;- when two scaffolds are linked by </span><span class="c38">at least</span><span class="c8 c9">&nbsp;two independent BAC ends.</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c38">possible misassembly </span><span class="c8 c9">- that should be investigated because BAC ends are tipically a few, and a lack of coverage could be just because of this.</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">probable misassemblies - although usually highly variable in insert size, it is possible to spot misassembled or critical regions when its necessary to supposean excess of insert.</span></li></ul><p class="c66 c23"><span class="c8 c9"></span></p><h2 class="c20" id="h.iiog2b8zahul"><span class="c22 c9">Convert chromatograms to FASTA files</span></h2><p class="c0"><span>If no other tools are available, we provide a simple Perl script (abi2fa.pl) that performs this step using ABI.pm module (</span><span class="c3"><a class="c15" href="https://www.google.com/url?q=http://search.cpan.org/~malay/ABI-1.0/ABI.pm&amp;sa=D&amp;source=editors&amp;ust=1666000537246129&amp;usg=AOvVaw1EgO7b1xWQssq904HcHBwx">http://search.cpan.org/~malay/ABI-1.0/ABI.pm</a></span><span class="c8 c9">). </span></p><h2 class="c20" id="h.bma29q4o2vk4"><span class="c22 c9">Aligning BAC ends</span></h2><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 517.00px; height: 262.00px;"><img alt="" src="images/image6.png" style="width: 517.00px; height: 262.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c23 c66"><span class="c9 c38 c48 c59 c46 c83"></span></p><p class="c66"><span class="c38 c46">Figure 8</span><span class="c41 c9">&nbsp;- BAC ends pipeline workflow.</span></p><p class="c18"><span class="c8 c9"></span></p><p class="c0"><span>The </span><span class="c2">align_bacends.pl</span><span class="c8 c9">&nbsp;script requires BLAST binaries (blastall) and a formatted database of scaffolds.</span></p><p class="c0"><span class="c8 c9">This is the a typical usage:</span></p><p class="c0 c33"><span class="c26 c9">align_bacends.pl fa FOR REV scaffolds.fna&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p><p class="c0"><span class="c8 c9">The four mandatory parameters are:</span></p><p class="c0"><span class="c8 c9">1) the extension of fasta files with BAC ends sequences (one per file). You have to launch the program from the directory containing them;</span></p><p class="c0"><span class="c8 c9">2-3) The string to classify &ldquo;For&rdquo; and &ldquo;Rev&rdquo; sequences, that has to be present in the file name;</span></p><p class="c0"><span class="c8 c9">4) The BLAST database (scaffolds).</span></p><p class="c0"><span class="c8 c9">The output is a report of validated regions, a list of possible scaffold to be joined and a list of critical regions that deserve further investigations.</span></p><p class="c0"><span class="c8 c9">Finally a report of how many scaffolds (and Mbp) are confirmed. We report a subset of the output here:</span></p><h3 class="c43" id="h.t9wjha73xkb4"><span class="c49 c38 c40 c48">Possible scaffold joining</span></h3><p class="c0"><span class="c38 c46 c55">Tag &nbsp; Starting Scaffold &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Ending Scaffold&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BACs&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Notes</span></p><p class="c0"><span class="c26 c9">&nbsp;[?] contig01469 +&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- contig00608&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.</span></p><p class="c0"><span class="c26 c9">&nbsp;[?] Scaffoldcontig00345 +&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+ contig00225&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.</span></p><p class="c0"><span class="c26 c9">&nbsp;[?] Scaffoldcontig00147 -&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+ Scaffoldcontig00282&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.</span></p><p class="c0"><span class="c26 c9">&nbsp;[?] contig00323 -&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+ Scaffoldcontig01068&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.</span></p><p class="c0"><span class="c26 c9">&nbsp;[?] contig00337 +&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+ contig00363&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.</span></p><p class="c0"><span class="c26 c9">&nbsp;[?] contig00273 +&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+ contig00493&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.</span></p><p class="c0"><span class="c26 c9">&nbsp;[?] contig01048 +&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- contig02066&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.</span></p><p class="c0"><span class="c26 c9">&nbsp;[?] contig00466 -&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+ contig02211&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><h3 class="c43" id="h.wxc96xdwdozs"><span class="c38 c40 c48 c49">Confirmed scaffolds</span></h3><p class="c0"><span class="c8 c9">Each confirmed sequence has a header line plus a line for each BAC pair confirming it. A couple of examples:</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c32 c9 c48 c7 c46">contig00022&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;568905&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;14&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;From: 96437&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;To: 552573&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;131694</span></p><p class="c0"><span class="c26 c9">seq_P2_Q3_-B04.fa&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;138655&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from:478917&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;to:340262&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+</span></p><p class="c0"><span class="c26 c9">seq_P1_Q1_-G02.fa&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;184511&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from:549922&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;to:365411&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+</span></p><p class="c0"><span class="c9 c26">seq_P1_Q2_-F01.fa&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;153533&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from:325597&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;to:479130&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-</span></p><p class="c0"><span class="c26 c9">seq_P2_Q4_-D11.fa&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;230809&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from:281509&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;to:512318&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-</span></p><p class="c0"><span class="c26 c9">seq_P1_Q4_-H01.fa&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;71293&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from:96437&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;to:167730&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-</span></p><p class="c0"><span class="c26 c9">&hellip;</span></p><p class="c0 c23"><span class="c26 c9"></span></p><p class="c0"><span class="c32 c9 c7 c46 c48">Scaffoldcontig00282&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;19397&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;From: 7884&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;To: 15535&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;844.5</span></p><p class="c0"><span class="c26 c9">seq_P2_Q3_-H06.fa&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;658&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from:8542&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;to:7884&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-</span></p><p class="c0"><span class="c26 c9">seq_P1_Q2_-D07.fa&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1031&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from:14504&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;to:15535&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+</span></p><p class="c0 c23"><span class="c8 c9"></span></p><h3 class="c43" id="h.i3uewm1jt28e"><span class="c49 c38 c40 c48">Summary log</span></h3><p class="c0 c23"><span class="c8 c9"></span></p><p class="c16 c25"><span class="c26 c9">&nbsp;FOR files &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;665 (FOR)</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;REV files &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;665 (REV)</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;Other (skip) &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1</span></p><p class="c16 c25 c23"><span class="c26 c9"></span></p><p class="c16 c25"><span class="c26 c9">&nbsp; &nbsp;Paired &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;665</span></p><p class="c16 c25"><span class="c26 c9">&nbsp; &nbsp;REV not f. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0</span></p><p class="c16 c25 c23"><span class="c26 c9"></span></p><p class="c16 c25"><span class="c26 c9">&nbsp; &nbsp;Failed seqs.&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;27</span></p><p class="c16 c25"><span class="c26 c9">&nbsp; &nbsp;NoMatch &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;119</span></p><p class="c16 c25"><span class="c26 c9">&nbsp; &nbsp;Confirmed &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;278 </span></p><p class="c16 c25"><span class="c26 c9">&nbsp; &nbsp; &nbsp; Bp Covered &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;11527770 bp</span></p><p class="c16 c25"><span class="c26 c9">&nbsp; &nbsp;Connections &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;203</span></p><p class="c0 c23"><span class="c8 c9"></span></p><h3 class="c43" id="h.tesarwftwyz4"><span class="c49 c38 c40 c48">Graphical Output </span></h3><p class="c0"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 1153.00px; height: 183.00px;"><img alt="" src="images/image4.png" style="width: 1153.00px; height: 183.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><h1 class="c12" id="h.xnj70ct6f8p"><span class="c6"></span></h1><p class="c0"><span class="c8 c9">For each scaffold the red bar represents the scaffold itself, having the shaded vertical lines representing the gap between a contig and the following one.</span></p><p class="c0"><span>The red arcs represent BAC ends. The green line is the covered region in terms of contigs.</span></p><h1 class="c12" id="h.upsu5zmu4vsk"><span class="c6"></span></h1><hr style="page-break-before:always;display:none;"><h1 class="c12" id="h.y3g74vas5faf"><span class="c6"></span></h1><h1 class="c20" id="h.cgmlf9vw9d0v"><span class="c6">Manual refinements</span></h1><p class="c0"><span class="c8 c9">Several source of information, including BAC ends as described in previous chapter or telomeric extremities marked, can be used to improve the scaffolding.</span></p><h2 class="c20" id="h.wdnpmskf72hx"><span class="c22 c9">Try exploring MP graph</span></h2><p class="c0"><span class="c8 c9">If two scaffolds are supposed to be connected you can find the facing contigs and try to see if they are connected via MP through some other contigs.</span></p><p class="c0"><span>The </span><span class="c2">contig_from_to.pl</span><span class="c8 c9">&nbsp;scripts will create, if possible, a scaffold joining the two contigs.</span></p><h2 class="c20" id="h.73f4uoljz4uz"><span class="c22 c9">Join scaffolds</span></h2><p class="c0"><span class="c8 c9">join_scaffolds.pl is a script to create a super scaffolds. This is the syntax:</span></p><p class="c0 c23 c33"><span class="c26 c9"></span></p><p class="c0 c33"><span class="c26 c9">join_scaffolds.pl newname &nbsp;scaffold1C scaffold2U scaffold3C..</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">Where &ldquo;newname&rdquo; is the name of the superscaffold, followed by a list of scaffolds names with their orientation expressed as &ldquo;U&rdquo; for &ldquo;uncomplemented&rdquo; or &ldquo;C&rdquo; for complemented, if it should be reversed complemented.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c19 c9"></span></p><hr style="page-break-before:always;display:none;"><p class="c0 c23"><span class="c19 c9"></span></p><h1 class="c79" id="h.b2bmvmcf2p29"><span class="c6">Appendix I &ndash; Core scripts</span></h1><p class="c0"><span class="c9 c38 c48 c59 c78">Core Scripts</span></p><a id="t.98c380a9d14dc4dba230f79e487a2ec05203a65d"></a><a id="t.1"></a><table class="c53"><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span>scampi_project.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c0"><span>Create a new database with a dedicated MySQL user.</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span>scampi_testconfig.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span>Test a database connection using a configuration file created by scampi_project.pl</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">contig_to_db.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Parses a multifasta file with contig and prints a SQL script to create and populate the contigs table. The table will lack coverage information if not available in the sequence headers.</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">scampi_mates_to_arcs.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Reads paired alignments to produce arcs (connections) between contigs.</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">scampi_arcs2sql.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Loads arcs file made by reads_to_arcs.pl into a SQL table.</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span>divorce.pl; divorce-enhanced.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span>Scripts for scaffolding from single seed. Used by the interface and by multi-scampi.pl core module.</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">multi-scampi.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">(in /web/inc) performs scaffolding for the whole genome, and eventually updates the database saving scaffolds produced.</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16 c23"><span class="c8 c9"></span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16 c23"><span class="c8 c9"></span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16 c23"><span class="c8 c9"></span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16 c23"><span class="c8 c9"></span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16 c23"><span class="c8 c9"></span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16 c23"><span class="c8 c9"></span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c78 c9 c38 c48 c59">Other tools</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16 c23"><span class="c8 c9"></span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">csfasta_to_fastq</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Combine reads and qualities in a single FASTQ file</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">pass</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span>Alignment tool (Campagna</span><span class="c7">&nbsp;et al.</span><span class="c8 c9">&nbsp;2007)</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">pass_pair</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Combine the alignments of the two pairs/mates</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">pair_mates.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span class="c8 c9">Read the two fasta/fastq files of mate reads, and prints only the reads present in both. Useful for sequencing machines that gives as output also orphan reads.</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span>pairin_simplifier.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span>Converts UNIQUE_PAIR from pass_pair in a sorted file</span></p></td></tr><tr class="c37"><td class="c1" colspan="1" rowspan="1"><p class="c16"><span>consistency.pl</span></p></td><td class="c42" colspan="1" rowspan="1"><p class="c16"><span>Breaks apart possible misassemblies</span></p></td></tr></table><p class="c0 c23"><span class="c19 c9"></span></p><hr style="page-break-before:always;display:none;"><p class="c0 c23"><span class="c19 c9"></span></p><h1 class="c20" id="h.1qo7lg6044t6"><span class="c6">Appendix II &ndash; Paired alignments</span></h1><p class="c0"><span>A pivotal step in scaffolding with paired-ends / mate-paired reads is to properly join the two sequences originated by a single DNA fragment (Figure A2.1). We refer to this step as &ldquo;</span><span class="c38">pairing</span><span class="c8 c9">&rdquo;. There are two main approaches to perform pairing:</span></p><ol class="c30 lst-kix_xzg29ra0f9c3-0 start" start="1"><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Aligning separately the two set of reads (&ldquo;forward&rdquo; and &ldquo;reverse&rdquo;) and then joining the results.</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Aligning the two set of reads together thanks to &ldquo;paired-reads&rdquo; aware aligners.</span></li></ol><p class="c0"><span class="c8 c9">For SOLiD Mate Paired reads we recommend the first strategy, using PASS as aligner and the pairing tool it provides to join the two alignments.</span></p><p class="c0"><span class="c8 c9">For base-space encoded reads (Illumina, Ion Torrent etc) both strategies can be used; if you choose the latter, ScaMPI has been tested with Bowtie2.</span></p><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 420.00px; height: 117.00px;"><img alt="" src="images/image3.png" style="width: 420.00px; height: 117.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c61">Figure A2.1 - When aligning mate paired reads against a reference, they have the same orientation (</span><span class="c61 c7">i.e. </span><span class="c61">strand). It&rsquo;s important to remember tha F3 mate is always upstream, while R3 is downstream. Paired end reads, on the other hand, align on opposite strands.</span></p><h2 class="c20" id="h.7finujinv245"><span class="c22 c9">Aligning and pairing SOLiD Mate Paired reads</span></h2><p class="c0"><span class="c8 c9">SOLiD Mate Paired reads (see Figure A2.1, left) are usually 50 or 60 bp reads with F3 and R3 tags, respectively. </span></p><p class="c0"><span class="c8 c9">Alignment can be performed with PASS using this command:</span></p><p class="c0 c33"><span class="c14 c9">pass -csfastq library1_F3.csfastq -d contigs.fna \</span></p><p class="c0 c33"><span class="c14 c9">-p 11111100111111 -cpu 12 -fid 90 -uniq -gff &gt; lib1F3.gff</span></p><p class="c0"><span>The </span><span class="c7">-uniq</span><span>&nbsp;switch prints only unique alignments, and it&rsquo;s recommended because ScaMPI will use only those kind of aligned reads. It is not mandatory, though, so if you performed alignments for other purposes, you can still use the GFF file produced.</span><hr style="page-break-before:always;display:none;"></p><a id="id.93kod45arhxj"></a><h1 class="c20" id="h.imol8vsh039"><span class="c6">Appendix III &ndash; Contig correctness check</span></h1><p class="c0 c23"><span class="c29 c9"></span></p><p class="c0"><span class="c8 c9">Mate Paired reads with both mates aligned inside the same contig can be used as an indicator of mis-assemblies under the assumption that the physical coverage should be uniform in a properly assembled contig.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 588.00px; height: 72.00px;"><img alt="" src="images/image8.png" style="width: 588.00px; height: 72.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c17">Figure A3.1</span><span class="c29 c9">&nbsp;- Scheme depicting physical coverage. </span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">A major limit of this approach is that only contigs larger than the average insert size of mate pairs can be verified. Moreover - especially with short insert size libraries - particularly complex regions could lead to a drop of physical coverage not connected with mis-assemblies (i.e. false positives are possible).</span></p><p class="c0"><span class="c8 c9">Nevertheless breaking apart contigs showing a drop of physical coverage can certainly benefits the whole scaffolding process.</span></p><p class="c0"><span>First, to speedup the process, the UNIQUE_PAIR file produced by pass_pair (see </span><span class="c3"><a class="c15" href="#id.rc7ixubsbjat">Pairing alignments</a></span><span>) has to be converted to a slimmer format and sorted with the </span><span class="c2">pairin_simplifier.pl</span><span class="c8 c9">&nbsp;script, as shown below:</span></p><p class="c0 c33"><span class="c56 c9 c48 c46">pairin_simplifier.pl -i UNIQUE_PAIR -gff -sort pairs.txt</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">The -gff parameter (requiring no arguments) specifies that the UNIQUE_PAIR is in GFF format, while default input is SAM.</span></p><p class="c0"><span class="c8 c9">By default the script would convert the file to STDOUT, in a tabular format that has to be sorted to be used by correctness checking module.</span></p><p class="c0"><span>The </span><span class="c7">-sort outputfile</span><span class="c8 c9">&nbsp;parameter will save output in a temporary file and at the end sort it (using UNIX sort command, that is thus required).</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>The output can be parsed with the </span><span class="c2">correctness.pl</span><span class="c8 c9">&nbsp;script:</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;+--------------------------------------------------------------------+</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;| &nbsp;PHYSICAL COVERAGE ANALYSIS &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;v. 2.00 &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CRIBI 2013 &nbsp; &nbsp;|</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;+--------------------------------------------------------------------+</span></p><p class="c16 c25 c23"><span class="c26 c9"></span></p><p class="c16 c25"><span class="c32 c46">&nbsp;-i &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;file &nbsp; Unique Pair </span><span class="c38 c55 c46 c73">sorted</span><span class="c26 c9">&nbsp;(produced by pairin_simplifier.pl)</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;-o &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;dir &nbsp; &nbsp;Output directory for tracks (default: current)</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;-t &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Print all tracks (contigname.track)</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;-bt &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Print tracks of broken contigs/sequence</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;-sam&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;file &nbsp; SAM file to read header</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;-ref&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;file &nbsp; Reference (no need to provide -sam)</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;-save file &nbsp; Print splitted contigs to file (requires -ref)</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;-cov&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;int &nbsp; &nbsp;Minimum physical coverage of contigs to be printed</span></p><p class="c16 c25"><span class="c26 c9">&nbsp;-len&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;int &nbsp; &nbsp;Minimum length of contigs to be printed</span></p><p class="c0"><span class="c9 c48 c46 c56">&nbsp;</span></p><p class="c0"><span class="c8 c9">A typical usage:</span></p><p class="c0"><span class="c46">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c64 c46">correctness.pl -i pairs.txt -ref contigs.fa -save new_contigs.fa -len 100</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">A graphical representation of each contig physical coverage can be created with the track2img.pl script (requires GD module). An example of output is show in the picture below.</span></p><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 503.00px; height: 61.00px;"><img alt="" src="images/image9.png" style="width: 503.00px; height: 61.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c17">Figure A3.1 </span><span class="c61">- Physical coverage of a contig indicate a probable misassemby.</span><hr style="page-break-before:always;display:none;"></p><a id="id.9kk3x3bk63v"></a><h1 class="c20" id="h.jp00ill4hipa"><span class="c6">Appendix IV - Telomere identification </span></h1><p class="c0"><span>The &ldquo;</span><span class="c38">Telomeric Repeat Analysis Pipeline</span><span>&rdquo; (</span><span class="c38">TRAP</span><span>) has been developed to enhance scaffolding by identifying chromosomal ends. In the test case used for ScaMPI, the microalga </span><span class="c7">N. gaditana</span><span class="c8 c9">, little was known about its genomic arrangement and discovering ~60 telomeres let to an estimate of 30 chromosomes. This number is quite important to understand the status of scaffolding and how to improve it.</span></p><p class="c0"><span>A version of TRAP tightly integrated with the ScaMPI web interface is under development and will be released with </span><span class="c7">ScaMPI v.2.0</span><span class="c8 c9">.</span></p><p class="c0"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 600.00px; height: 156.45px;"><img alt="" src="images/image2.png" style="width: 600.00px; height: 156.45px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><h2 class="c20" id="h.twlbboe8ouo2"><span class="c22 c9">TRAP overview</span></h2><ol class="c30 lst-kix_7b6g4smvhj31-0 start" start="1"><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Identification of the telomeric repeat (or a set of candidate repeats), if not known;</span></li><li class="c0 c25 c28 li-bullet-0"><span>Isolation of MP reads having one of the mates filled with the telomeric repeat (at lest </span><span class="c7">n</span><span class="c8 c9">-times);</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Mapping of the isolated MP against scaffold and/or contigs;</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Analysis of the alignment: if the alignments are widespread the candidate repeat is probably wrong, if they concentrate at the end of contigs they are likely correct. If they concentrate at the end of scaffolds both the repeat and the scaffold should be correct;</span></li></ol><h2 class="c20" id="h.4zvtntx2qnm9"><span class="c7">De novo</span><span class="c22 c9">&nbsp;identification of the repeat </span></h2><p class="c0"><span>In the unfortunate case of unknown telomeric repeat, it has to be guessed using known repeats or discovered </span><span class="c7">de novo</span><span>. It might be the case that analysing longer reads (454, Torrent or even Illumina&rsquo;s) we can identify unexpectedly over-represented motifs, and after discarding low complexity repeats (like, </span><span class="c32">CTCTCTCT...</span><span class="c8 c9">) we can keep the remaining and test them. </span></p><p class="c0"><span class="c8 c9">For this step many programs can be used. We provide a simple script (trap_words.pl) that can be used for this purpose.</span></p><p class="c0"><span class="c8 c9">Usage:</span></p><p class="c0 c33"><span class="c32 c45">trap_words.pl -i </span><span class="c32 c45 c7">reads_file</span><span class="c32 c45">&nbsp;-w </span><span class="c32 c45 c7">word_len</span><span class="c32 c45">&nbsp;-m </span><span class="c32 c45 c7">min_read_len</span><span class="c9 c11">&nbsp;-filter \</span></p><p class="c0 c33"><span class="c11 c9">&gt; reads.fa 2&gt; log.txt</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">This script will produce as output a set of reads with tandem repeats, and a &ldquo;log&rdquo; with a set of most frequently found motifs (note that the printed motif is the first version in alphabetical order so that TAACC becomes AACCT).</span></p><p class="c0"><span>Last lines of &ldquo;log&rdquo; file from </span><span class="c7">N. gaditana</span><span class="c8 c9">&nbsp;reads:</span></p><p class="c34 c25 c23"><span class="c26 c9"></span></p><p class="c34 c25"><span class="c26 c9">ACCTGC &nbsp;79</span></p><p class="c34 c25"><span class="c26 c9">ACACAT &nbsp;81</span></p><p class="c34 c25"><span class="c26 c9">ATATCT &nbsp;90</span></p><p class="c34 c25"><span class="c26 c9">AGCAGC &nbsp;169</span></p><p class="c34 c25"><span class="c26 c9">AACAGC &nbsp;237</span></p><p class="c34 c25"><span class="c26 c9">AGGTAT &nbsp;259</span></p><p class="c34 c25"><span class="c26 c9">AGGTGT &nbsp;261</span></p><p class="c25 c34"><span class="c26 c9">AGAGGT &nbsp;267</span></p><p class="c34 c25"><span class="c26 c9">AGGGGT &nbsp;288</span></p><p class="c34 c25"><span class="c26 c9">AACCCT &nbsp;393</span></p><p class="c34 c25"><span class="c32 c46">AGGGTT &nbsp;930&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# this TTAGGG is the telomeric tandem repeat</span></p><h2 class="c20" id="h.lf5orxq9tzze"><span class="c22 c9">Isolation of motif-containing MP reads</span></h2><p class="c0"><span>First we should convert the two MP files into a single file for faster search with the </span><span class="c2">trap_preparereads.pl </span><span class="c8 c9">program used as follows:</span></p><p class="c0 c33"><span class="c32 c45">trap_preparereads.pl </span><span class="c32 c45 c7">file_F3.csfastq file_R3.csfastq </span><span class="c32 c45">&gt; </span><span class="c32 c9 c48 c45 c7">file.mp1</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">The output is a tabular file with this structure (suggest:</span></p><p class="c0 c33"><span class="c19 c9">Mate_name&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;F3_seq&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;F3_qual&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;R3_seq&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;R3_qual</span></p><p class="c0 c23 c33"><span class="c19 c9"></span></p><p class="c0"><span>This can easily parsed and used to reconstruct paired reads in FASTQ format using the </span><span class="c2">trap_PatternSelector.pl </span><span class="c8 c9">script as described below:</span></p><p class="c0"><span class="c32">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;trap_PatternSelector.pl </span><span class="c32 c7">file.mp1 pattern -single </span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span>The pattern has to be in color space, for example </span><span class="c32">03200</span><span>&nbsp;instead of </span><span class="c32">TTAGGG</span><span>. The program will save two output files adding to the input file name the </span><span class="c32">.F3</span><span>&nbsp;and </span><span class="c32">.R3</span><span class="c8 c9">&nbsp;suffixes by default. Using the -single switch causes the whole dataset (F3 and R3) to be printed in the same file (.F3).</span></p><h2 class="c20" id="h.jsn1n3mzj9cd"><span class="c22 c9">Alignment of selected reads</span></h2><p class="c0"><span class="c8 c9">PASS with default parameters can be used to align reads against contigs and/or scaffolds.</span></p><p class="c0"><span class="c45">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c9">pass -csfastq selected_reads.F3 -d contigs.fna -gff -cpu 12 &gt; aln.gff</span></p><p class="c0 c23"><span class="c19 c9"></span></p><p class="c0"><span>It&rsquo;s important to specify the </span><span class="c7">-gff</span><span class="c8 c9">&nbsp;switch, as downstream analysis is based on this format.</span></p><h2 class="c20" id="h.sr4hf46m73j"><span class="c22 c9">Extraction of putative peri-telomeric regions</span></h2><p class="c0"><span>A simple scripts evaluates the density of alignments. It&rsquo;s called </span><span class="c2">trap_gff.pl</span><span>&nbsp;and its arguments are: the alignment file in GFF format, the reference, the window and the enrichment of alignments requested to mark a region as putative telomere.</span></p><p class="c0 c33"><span class="c32 c61">trap_gff</span><span class="c14 c9">.pl newscaf.gff test.scaffolds.fna 400 10 &nbsp;&gt; telomericregions.txt</span></p><p class="c0"><span class="c8 c9">The output is a tabular file with these fields</span></p><p class="c0 c33"><span class="c14 c9">Ref_name&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Position(window)&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;N_aln_for&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;N_aln_rev&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Flag&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Enrichment</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">Where the position refers to a windows as specified to the program and &ldquo;flag&rdquo; can be &ldquo;T&rdquo; for telomere or blank, and it&rsquo;s T when the enrichment of alignments in the windows is greater than the threshold specified in the parameters. An example for a single scaffold is shown below:</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c16 c25"><span class="c14 c9"># ------ Seq:00534 (93880 bp, 12458 alignments)</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 0 &nbsp; &nbsp; &nbsp; 60 &nbsp; &nbsp; &nbsp;119 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 1.3X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 1000 &nbsp; &nbsp;1724 &nbsp; &nbsp;2954 &nbsp; &nbsp;T &nbsp; &nbsp; &nbsp; 35.3X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 2000 &nbsp; &nbsp;2167 &nbsp; &nbsp;3305 &nbsp; &nbsp;T &nbsp; &nbsp; &nbsp; 41.2X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 3000 &nbsp; &nbsp;413 &nbsp; &nbsp; 896 &nbsp; &nbsp; T &nbsp; &nbsp; &nbsp; 9.9X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 4000 &nbsp; &nbsp;119 &nbsp; &nbsp; 201 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 2.4X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 5000 &nbsp; &nbsp;30 &nbsp; &nbsp; &nbsp;11 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;0.3X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 6000 &nbsp; &nbsp;7 &nbsp; &nbsp; &nbsp; 4 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.1X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 7000 &nbsp; &nbsp;5 &nbsp; &nbsp; &nbsp; 3 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.1X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 8000 &nbsp; &nbsp;16 &nbsp; &nbsp; &nbsp;12 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;0.2X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 9000 &nbsp; &nbsp;7 &nbsp; &nbsp; &nbsp; 6 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.1X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 10000 &nbsp; 1 &nbsp; &nbsp; &nbsp; 4 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 11000 &nbsp; 6 &nbsp; &nbsp; &nbsp; 3 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.1X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 12000 &nbsp; 4 &nbsp; &nbsp; &nbsp; 5 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.1X</span></p><p class="c16 c25"><span class="c9 c14">00534 &nbsp; 13000 &nbsp; 3 &nbsp; &nbsp; &nbsp; 2 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c32 c9 c48 c61 c7">...cut...</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 77000 &nbsp; 0 &nbsp; &nbsp; &nbsp; 0 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 78000 &nbsp; 1 &nbsp; &nbsp; &nbsp; 5 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 79000 &nbsp; 2 &nbsp; &nbsp; &nbsp; 2 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 80000 &nbsp; 3 &nbsp; &nbsp; &nbsp; 0 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 81000 &nbsp; 3 &nbsp; &nbsp; &nbsp; 3 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 82000 &nbsp; 0 &nbsp; &nbsp; &nbsp; 0 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 83000 &nbsp; 5 &nbsp; &nbsp; &nbsp; 6 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.1X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 84000 &nbsp; 2 &nbsp; &nbsp; &nbsp; 4 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 85000 &nbsp; 3 &nbsp; &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 86000 &nbsp; 11 &nbsp; &nbsp; &nbsp;5 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.1X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 87000 &nbsp; 1 &nbsp; &nbsp; &nbsp; 3 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 88000 &nbsp; 1 &nbsp; &nbsp; &nbsp; 3 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 89000 &nbsp; 2 &nbsp; &nbsp; &nbsp; 3 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 90000 &nbsp; 1 &nbsp; &nbsp; &nbsp; 2 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 91000 &nbsp; 2 &nbsp; &nbsp; &nbsp; 0 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 92000 &nbsp; 1 &nbsp; &nbsp; &nbsp; 3 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25"><span class="c14 c9">00534 &nbsp; 93000 &nbsp; 1 &nbsp; &nbsp; &nbsp; 3 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 0.0X</span></p><p class="c16 c25 c23"><span class="c14 c9"></span></p><p class="c16 c23"><span class="c14 c9"></span></p><p class="c0"><span>You can see that there is a telomere at the begin (5&rsquo;) of the scaffold, while there is none at the end. The background noise is generally very low. After inspecting the file you can run the </span><span class="c2">extract_telomeres.pl</span><span class="c8 c9">&nbsp;script that find peaks and report telomeres when present at the begin (5&rsquo;), end (3&rsquo;) or non-canonical position (can be used to spot misassemblies).</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c38">Update</span><span class="c8 c9">: the new version (v. 1.1) of the script will print a more informative output:</span></p><p class="c16"><span class="c11 c40 c81"># ------ Seq:chr19 (59063183 bp, 126 alignments)<br>chr19 0 &nbsp; &nbsp; &nbsp; &nbsp;0.00% &nbsp; 52 &nbsp;T &nbsp;0.8X<br>chr19 10000 &nbsp; &nbsp;0.02% &nbsp; 0 &nbsp; &nbsp; &nbsp;0.0X<br>chr19 20000 &nbsp; &nbsp;0.03% &nbsp; 0 &nbsp; &nbsp; &nbsp;0.0X<br>chr19 30000 &nbsp; &nbsp;0.05% &nbsp; 0 &nbsp; &nbsp; &nbsp;0.0X<br>chr19 40000 &nbsp; &nbsp;0.07% &nbsp; 0 &nbsp; &nbsp; &nbsp;0.0X<br>...<br>chr19 59020000 99.93% &nbsp;0 &nbsp; &nbsp; &nbsp;0.0X<br>chr19 59030000 99.94% &nbsp;17 &nbsp; &nbsp; 0.3X<br>chr19 59040000 99.96% &nbsp;3 &nbsp; &nbsp; &nbsp;0.0X<br>chr19 59050000 99.98% &nbsp;50 &nbsp; T 0.8X<br>chr19 59060000 99.99% &nbsp;0 &nbsp; &nbsp; &nbsp;0.0X</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">While comment lines starting with a sharp still exist, they are no longer necessary as the reference name is reported in column 1.</span></p><p class="c0"><span class="c8 c9">Column 2 still keeps the position of the windows, but now column 3 reports the position as a percentage to make it easier to retrieve extremities.</span></p><p class="c0"><span class="c8 c9">Column 4 reports alignment (For + Rev, while former output reported the two separately).</span></p><p class="c0 c23"><span class="c8 c9"></span></p><h1 class="c12" id="h.a3u5g8wc99p9"><span class="c6"></span></h1><hr style="page-break-before:always;display:none;"><h1 class="c12" id="h.mpf8ek8s2f3g"><span class="c6"></span></h1><a id="id.it6whu3pmxr3"></a><h1 class="c20" id="h.80sg42ok1843"><span class="c6">Appendix V - Default Plug-ins</span></h1><h2 class="c20" id="h.xxhaxxlkd773"><span class="c22 c9">Installing new plug ins</span></h2><p class="c0"><span>Plug-ins are supplied as an archive to be extracted within the </span><span class="c64">/web/inc/plugins </span><span>directory of ScaMPI. Each plugins has a </span><span class="c7">_pluginname.php</span><span>&nbsp;file and (at least) a </span><span class="c7">pluginname.php</span><span>&nbsp;file, the former being a shortcut for the top bar menu, the latter being the plugin </span><span class="c7">per se</span><span class="c8 c9">. </span></p><p class="c0"><span>Plugins can be displayed singularly or grouped, and this settings is stored in the </span><span class="c7">_pluginname.php</span><span class="c8 c9">&nbsp;file.</span></p><p class="c0"><span class="c8 c9">Default plugins supplied with scampi are:</span></p><ol class="c30 lst-kix_cua9hof9g9u-0 start" start="1"><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Backup DB</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">BLAST</span></li><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Make AGP</span></li></ol><h2 class="c20" id="h.s9zfsrysg86k"><span class="c22 c9">Backup DB</span></h2><p class="c0"><span class="c8 c9">This plugins takes a snapshot of the whole database, printing the output to the browser or into a file. Default output directory is /tmp/ being writeable by (most) web servers, but can be changed. The intended use is to move/copy the db file into an appropriate location/device.</span></p><h2 class="c20" id="h.2x7axij1wj9c"><span class="c22 c9">BLAST</span></h2><p class="c0"><span class="c8 c9">Requirements / Preliminary steps:</span></p><ol class="c30 lst-kix_o260vi9n67px-0 start" start="1"><li class="c0 c25 c28 li-bullet-0"><span class="c8 c9">Install BLAST. The &ldquo;formatdb&rdquo; and &ldquo;blastall&rdquo; binaries are required.</span></li><li class="c0 c25 c28 li-bullet-0"><span>Copy (or symlink) the contigs.fa file in the</span><span class="c64">&nbsp;/plugins/blast</span><span class="c8 c9">&nbsp;subdirectory.</span></li><li class="c0 c25 c28 li-bullet-0"><span>Format the database: </span><span class="c64">formatdb -p F -i contigs.fa</span><span>&nbsp;(from </span><span class="c64">/plugins/blast</span><span class="c8 c9">)</span></li></ol><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">Once that these steps are completed, accessing the &ldquo;BLAST&rdquo; plugin will show you a form where to paste the sequence you want to BLAST and some options.</span></p><p class="c0"><span class="c8 c9">Note that putting more databases in the blast directory will enable you choosing the reference, having contigs.fa as the default (and required) one.</span></p><h2 class="c20" id="h.g5sqcot0aaq6"><span class="c22 c9">Make AGP</span></h2><p class="c0"><span class="c8 c9">Just click it from the home page to produce a whole genome scaffolding file in AGP format. If clicked from a contig, will produce the AGP only for its scaffold.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0"><span class="c8 c9">There are other plug-ins in our website that can be installed.</span></p><p class="c0"><span class="c8 c9">ORF Finder</span></p><p class="c0"><span class="c8 c9">Primer Picking</span></p><p class="c0 c23"><span class="c8 c9"></span></p><h1 class="c12" id="h.tx4dn9pol116"><span class="c6"></span></h1><hr style="page-break-before:always;display:none;"><h1 class="c12" id="h.zgo5j0b9ptf7"><span class="c6"></span></h1><h1 class="c20" id="h.165ipncd9vox"><span class="c6">Appendix VI - File formats</span></h1><h2 class="c20" id="h.m3jdp1ms1o5m"><span class="c22 c9">Input files</span></h2><p class="c0"><span class="c8 c9">Contigs have to be produced in FASTA format. ScaMPI requires to know the coverage of each contig, and because most assemblers store information about each contig in its header line, a program that parses the FASTA file is provided to extract sequence, its length and the coverage from the contigs file to populate a MySQL table (called &lsquo;contigs&rsquo;) having this structure:</span></p><p class="c0"><span class="c26 c9">CREATE TABLE contigs (</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;id&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;INT AUTO_INCREMENT PRIMARY KEY,</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;name&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;VARCHAR(255),&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--contig name as present in the FASTA file</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;len&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;INT,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--contig size as calculated from FASTA file</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;cov&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;FLOAT,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--contig coverage as extracted from FASTA </span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;scaffold &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;VARCHAR(200),&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--scaffold name the contigs belongs to</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;sid&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;INT,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--position in the scaffold (1=First...)</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;dir&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CHAR(1),&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--orientation in the scaffold (U or C)</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;notes&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;TEXT,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--user supplied notes,</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;t3&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BOOL,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--flanks a telomer3 at 3&rsquo;</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;t5&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BOOL,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--flanks a telomere at 5&rsquo;</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;exclude&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BOOL&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--exclude from scaffolding</span></p><p class="c0"><span class="c26 c9">);</span></p><h2 class="c20" id="h.2zbsvz8abfwc"><span class="c22 c9">Alignments of MP/PE: format and pairing</span></h2><p class="c0"><span>Reads are expected to be in FASTQ format, to be aligned against contigs using PASS (binary supplied or source available at </span><span class="c3"><a class="c15" href="https://www.google.com/url?q=http://pass.cribi.unipd.it/&amp;sa=D&amp;source=editors&amp;ust=1666000537289082&amp;usg=AOvVaw0d5MoezQx1VezphQTv4KNv">http://pass.cribi.unipd.it/</a></span><span class="c8 c9">). PASS supports various output format, but the preferred for ScaMPI is the GFF. A script to convert SAM to GFF is supplied in the package.</span></p><p class="c0"><span class="c8 c9">The two files (first and second mate/pair) are expected to be aligned independently, and then paired with the &ldquo;PASS PAIR&rdquo; program that classify pairs in categories (requires to know the library type and the library insert size), the two being used by ScaMPI being:</span></p><p class="c0 c25"><span class="c38">UNIQUE_PAIR</span><span class="c8 c9">&nbsp;- If the two mates/pairs align uniquely within the same reference contig with the proper mutual orientation and distance</span></p><p class="c0 c25"><span class="c38">UNIQUE_PAIR_OUT</span><span class="c8 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- If the two mates/pairs align uniquely within two different contigs, and the distance from the extremities is still compatible with the insert size.</span></p><p class="c0"><span class="c8 c9">The two files are again in GFF format, having the alignment for the first mate followed by the alignment of the second mate.</span></p><p class="c0"><span>For parsing convenience we provide a script to convert these files into a more compact and faster to read format, called &ldquo;</span><span class="c7">tabular reads file</span><span class="c8 c9">&rdquo; structured as follows:</span></p><p class="c0"><span class="c26 c9">a1_contig&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;a2_contig&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;a1_start&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;a2_start&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;a1_end&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;a2_end a1_strand&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;a2_strand</span></p><p class="c0"><span class="c8 c9">Where a1 and a2 refers to the alignment of the first and second pair, respectively.</span></p><h2 class="c20" id="h.v64zhypitvii"><span class="c22 c9">Arcs table</span></h2><p class="c0"><span class="c8 c9">The arcs table is structured as follows:</span></p><p class="c0"><span class="c26 c9">CREATE TABLE arcs (</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;id&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;INT AUTO_INCREMENT PRIMARY KEY,</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;lib&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;INT,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--library name</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;c1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;VARCHAR(200),&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--contig1 name</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;c2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;VARCHAR(200),&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--contig2 name</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;e1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;INT,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--end of contig1 facing contig2 (5 or 3)</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;e2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;INT,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--end of contig2 facing contig1 (5 or 3),</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;arcs&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;INT,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--number of alignments supporting this arc</span></p><p class="c0"><span class="c26 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;sigma&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;FLOAT&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--% of alignment supporting this orientation</span></p><p class="c0"><span class="c32 c46">);</span></p><p class="c0"><span class="c8 c9">The arcs table is populated by a script that parses UNIQUE_PAIR_OUT file and cluster connections between contigs, printing only those with a minimum number of alignments and minimum % of concordance (sigma) specified by the user.</span></p><p class="c0 c23"><span class="c8 c9"></span></p><h1 class="c12" id="h.13p7tyecr3se"><span class="c6"></span></h1><hr style="page-break-before:always;display:none;"><h1 class="c12" id="h.1igbxzktm93u"><span class="c6"></span></h1><h1 class="c20" id="h.lk49h9d6n9lh"><span class="c6">Appendix VII - Supplementary tools</span></h1><h2 class="c20" id="h.4u1mttywcfhx"><span class="c22 c9">Graphical plot of scaffolds (GraphViz)</span></h2><p class="c0"><span class="c8 c9">To inspect the connection between contigs in a scaffold, ScaMPI can print a graphical representation of the whole scaffold, as shown in the picture below.</span></p><p class="c0"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 615.00px; height: 93.00px;"><img alt="" src="images/image1.png" style="width: 615.00px; height: 93.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c41 c9">Figure A7.1 - Small portion of a scaffold picture. Each box represents a contig (not in scale), having both size and coverage labels. Arrows represent &ldquo;arcs&rdquo;. It is possible to find possible critical regions where, for example, only a connection support the link (of course this is normal if the contigs are larger than average insert size).</span></p><p class="c0 c23"><span class="c41 c9"></span></p><p class="c0"><span>The ScaMPI suite produces a script in &ldquo;dot&rdquo; language (</span><span class="c3"><a class="c15" href="https://www.google.com/url?q=http://www.graphviz.org/doc/info/lang.html&amp;sa=D&amp;source=editors&amp;ust=1666000537292279&amp;usg=AOvVaw2_SG5nungThBdJX1eRdCUI">http://www.graphviz.org/doc/info/lang.html</a></span><span class="c8 c9">) that can be converted to a picture by &ldquo;GraphViz&rdquo;. Please refer to &ldquo;dot&rdquo; manual to convert the dot script to a PDF document or bitmap image. An example of usage is:</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c33 c71"><span class="c11 c9">dot -Tpng scaffold_0123.dot &gt; scaffold_0123.png</span></p><p class="c71 c23"><span class="c11 c9"></span></p><p class="c0"><span class="c8 c9">To convert all .dot file to image it&rsquo;s simple to perform a loop in BASH as follows:</span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c71 c33"><span class="c11 c9">for i in *.dot; do dot -Tpng $i &gt; ${i/dot/png/}; done</span></p><p class="c0 c23"><span class="c8 c27"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c0 c23"><span class="c8 c9"></span></p><p class="c24"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 80.00px; height: 80.00px;"><img alt="" src="images/image11.gif" style="width: 80.00px; height: 80.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c24"><span class="c2">http://genomics.cribi.unipd.it/scampi/</span></p><div><p class="c24"><span>Page </span><span>&nbsp;of </span></p></div></body></html>